---
title: "BE Sensor analysis with Clojure: MDA231/HBES"
author: "Luke Dow & Henri Schmidt"
date: "11/16/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This is the pipeline fo the analysis of base editing sensor count data generated by bowtie alignment. This approach uses multiple reference alignments that iteratively measure C>T, C>A, C>G modification of target cytosines. Indels are calculated by subtracting all reads of the expected length (guide-scaffold-5'flank-target-3'flank), from the number of total reads that match the guide-scaffold-5'flank. 

First, load the required packages:

```{r echo=TRUE, message=FALSE, warning=FALSE}
library(dplyr)
library(tidyverse)
library(ggplot2)
library(viridis)
library(ggpubr)
library(cowplot)
library(gganimate)
library(readxl)
library(purrr)
library(stringr)

```

<p style="font-family: Arial; font-size:18pt; font-style:bold">
    Import transcript data
</p>
Import each of the tables containing alignment counts for each reference. The easiest way to do this is to set the working directory to the folder containing the relevant .txt files and import and rename columns using a loop:


```{r echo=TRUE, message=FALSE, warning=FALSE}
setwd("~/Box Sync/BE_sensor/MDA231_allPAM/results/")
# Create list of file names
filenames <- gsub("\\.csv$","", list.files(pattern="\\.csv$"))
# Run import for loop
for(i in filenames){
  assign(i, read.csv(paste(i, ".csv", sep="")))
}
```

Next, import the metadata ("variables") associated with each of the guide_IDs

```{r echo=TRUE, message=FALSE, warning=FALSE}
AP_variables <- read_csv("~/Box Sync/BE_sensor/Library_cloning/sensor_v3_20180824/allPAM_sensor_sequences_whitelist.xlsx", sheet = "variables")
```

<p style="font-family: Arial; font-size:18pt; font-style:bold">
    Process the count data
</p>
The raw count data for each replicate is merged into a single dataframe and normalized to the total read counts for each guide by calculating "precent editing" for each type of editing event. The mean of each replicate is calculated. Where needed, some duplicate guides are removed as they cannot be mapped accurately.

```{r echo=TRUE, message=FALSE, warning=FALSE}
# Calculate mean percentages for TARGET data
Cas9$percent_tCT = (Cas9$percent_tCT_REP1 + Cas9$percent_tCT_REP2)/2
Cas9$percent_tCTN = (Cas9$percent_tCTN_REP1 + Cas9$percent_tCTN_REP2)/2
Cas9$percent_tCAN = (Cas9$percent_tCAN_REP1 + Cas9$percent_tCAN_REP2)/2
Cas9$percent_tCGN = (Cas9$percent_tCGN_REP1 + Cas9$percent_tCGN_REP2)/2
Cas9$percent_indel = (Cas9$percent_indel_REP1 + Cas9$percent_indel_REP2)/2
Cas9$percent_editing = (Cas9$percent_tCTN + Cas9$percent_tCGN +
                                          Cas9$percent_tCAN + Cas9$percent_indel)
Cas9$percent_BE = (Cas9$percent_tCTN + Cas9$percent_tCGN +
                                          Cas9$percent_tCAN)
```

<p style="font-family: Arial; font-size:18pt; font-style:bold">
    Filter and merge the data
</p>
Data tables are filtered to retain only those guides that contain at least 100 reads in each of the replicates. Metadata ("variables") can be appended to the tables for later plotting purposes

```{r echo=TRUE, message=FALSE, warning=FALSE}
# Filter to 100+ reads/guide and remove replicate data
Cas9_filtered <- Cas9 %>% filter(total_REP1 > 49 & total_REP2 >49)
Cas9_mean <- Cas9_filtered[ ,-c(2:23)]

# Create separate mean file, rename columns, add variable and enzyme annoation
Cas9_mean_ann <- Cas9_filtered[ ,-c(2:23)]
Cas9_mean_ann <- list(Cas9_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
Cas9_mean_ann$enzyme <- "CAS9"
Cas9_mean_ann$cell_line <- "MDA231"

# Split by editing type
Cas9_any_editing <- Cas9_mean_ann[,-c(2,3,4,5,6,8)]
names(Cas9_any_editing)[2] <- "editing"
Cas9_any_editing$editing_type <- "all"
Cas9_tCT <- Cas9_mean_ann[,-c(3,4,5,6,7,8)]
names(Cas9_tCT)[2] <- "editing"
Cas9_tCT$editing_type <- "C>T_only"
Cas9_tCTN <- Cas9_mean_ann[,-c(2,4,5,6,7,8)]
names(Cas9_tCTN)[2] <- "editing"
Cas9_tCTN$editing_type <- "C>T"
Cas9_tCAN <- Cas9_mean_ann[,-c(2,3,5,6,7,8)]
names(Cas9_tCAN)[2] <- "editing"
Cas9_tCAN$editing_type <- "C>A"
Cas9_tCGN <- Cas9_mean_ann[,-c(2,3,4,6,7,8)]
names(Cas9_tCGN)[2] <- "editing"
Cas9_tCGN$editing_type <- "C>G"
Cas9_stack <- rbind(Cas9_tCT,Cas9_tCTN,Cas9_tCAN,Cas9_tCGN)
Cas9_stack <- Cas9_stack[complete.cases(Cas9_stack), ]

```

Repeat for each of the conditions (enzymes) included in the analysis

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Cas9NG
Cas9NG$percent_tCT = (Cas9NG$percent_tCT_REP1 + Cas9NG$percent_tCT_REP2)/2
Cas9NG$percent_tCTN = (Cas9NG$percent_tCTN_REP1 + Cas9NG$percent_tCTN_REP2)/2
Cas9NG$percent_tCAN = (Cas9NG$percent_tCAN_REP1 + Cas9NG$percent_tCAN_REP2)/2
Cas9NG$percent_tCGN = (Cas9NG$percent_tCGN_REP1 + Cas9NG$percent_tCGN_REP2)/2
Cas9NG$percent_indel = (Cas9NG$percent_indel_REP1 + Cas9NG$percent_indel_REP2)/2
Cas9NG$percent_editing = (Cas9NG$percent_tCTN + Cas9NG$percent_tCGN + Cas9NG$percent_tCAN + Cas9NG$percent_indel)
Cas9NG$percent_BE = (Cas9NG$percent_tCTN + Cas9NG$percent_tCGN + Cas9NG$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
Cas9NG_filtered <- Cas9NG %>% filter(total_REP1 > 49 & total_REP2 >49)
Cas9NG_mean <- Cas9NG_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
Cas9NG_mean_ann <- Cas9NG_filtered[ ,-c(2:23)]
Cas9NG_mean_ann <- list(Cas9NG_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
Cas9NG_mean_ann$enzyme <- "Cas9NG"
Cas9NG_mean_ann$cell_line <- "MDA231"
# Split by editing type
Cas9NG_any_editing <- Cas9NG_mean_ann[,-c(2,3,4,5,6,8)]
names(Cas9NG_any_editing)[2] <- "editing"
Cas9NG_any_editing$editing_type <- "all"
Cas9NG_tCT <- Cas9NG_mean_ann[,-c(3,4,5,6,7,8)]
names(Cas9NG_tCT)[2] <- "editing"
Cas9NG_tCT$editing_type <- "C>T_only"
Cas9NG_tCTN <- Cas9NG_mean_ann[,-c(2,4,5,6,7,8)]
names(Cas9NG_tCTN)[2] <- "editing"
Cas9NG_tCTN$editing_type <- "C>T"
Cas9NG_tCAN <- Cas9NG_mean_ann[,-c(2,3,5,6,7,8)]
names(Cas9NG_tCAN)[2] <- "editing"
Cas9NG_tCAN$editing_type <- "C>A"
Cas9NG_tCGN <- Cas9NG_mean_ann[,-c(2,3,4,6,7,8)]
names(Cas9NG_tCGN)[2] <- "editing"
Cas9NG_tCGN$editing_type <- "C>G"
Cas9NG_stack <- rbind(Cas9NG_tCT,Cas9NG_tCTN,Cas9NG_tCAN,Cas9NG_tCGN)
Cas9NG_stack <- Cas9NG_stack[complete.cases(Cas9NG_stack), ]

# RA2
RA2$percent_tCT = (RA2$percent_tCT_REP1 + RA2$percent_tCT_REP2)/2
RA2$percent_tCTN = (RA2$percent_tCTN_REP1 + RA2$percent_tCTN_REP2)/2
RA2$percent_tCAN = (RA2$percent_tCAN_REP1 + RA2$percent_tCAN_REP2)/2
RA2$percent_tCGN = (RA2$percent_tCGN_REP1 + RA2$percent_tCGN_REP2)/2
RA2$percent_indel = (RA2$percent_indel_REP1 + RA2$percent_indel_REP2)/2
RA2$percent_editing = (RA2$percent_tCTN + RA2$percent_tCGN + RA2$percent_tCAN + RA2$percent_indel)
RA2$percent_BE = (RA2$percent_tCTN + RA2$percent_tCGN + RA2$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
RA2_filtered <- RA2 %>% filter(total_REP1 > 49 & total_REP2 >49)
RA2_mean <- RA2_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
RA2_mean_ann <- RA2_filtered[ ,-c(2:23)]
RA2_mean_ann <- list(RA2_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
RA2_mean_ann$enzyme <- "RA2"
RA2_mean_ann$cell_line <- "MDA231"
# Split by editing type
RA2_any_editing <- RA2_mean_ann[,-c(2,3,4,5,6,8)]
names(RA2_any_editing)[2] <- "editing"
RA2_any_editing$editing_type <- "all"
RA2_tCT <- RA2_mean_ann[,-c(3,4,5,6,7,8)]
names(RA2_tCT)[2] <- "editing"
RA2_tCT$editing_type <- "C>T_only"
RA2_tCTN <- RA2_mean_ann[,-c(2,4,5,6,7,8)]
names(RA2_tCTN)[2] <- "editing"
RA2_tCTN$editing_type <- "C>T"
RA2_tCAN <- RA2_mean_ann[,-c(2,3,5,6,7,8)]
names(RA2_tCAN)[2] <- "editing"
RA2_tCAN$editing_type <- "C>A"
RA2_tCGN <- RA2_mean_ann[,-c(2,3,4,6,7,8)]
names(RA2_tCGN)[2] <- "editing"
RA2_tCGN$editing_type <- "C>G"
RA2_stack <- rbind(RA2_tCT,RA2_tCTN,RA2_tCAN,RA2_tCGN)
RA2_stack <- RA2_stack[complete.cases(RA2_stack), ]

# F2X
F2X$percent_tCT = (F2X$percent_tCT_REP1 + F2X$percent_tCT_REP2)/2
F2X$percent_tCTN = (F2X$percent_tCTN_REP1 + F2X$percent_tCTN_REP2)/2
F2X$percent_tCAN = (F2X$percent_tCAN_REP1 + F2X$percent_tCAN_REP2)/2
F2X$percent_tCGN = (F2X$percent_tCGN_REP1 + F2X$percent_tCGN_REP2)/2
F2X$percent_indel = (F2X$percent_indel_REP1 + F2X$percent_indel_REP2)/2
F2X$percent_editing = (F2X$percent_tCTN + F2X$percent_tCGN + F2X$percent_tCAN + F2X$percent_indel)
F2X$percent_BE = (F2X$percent_tCTN + F2X$percent_tCGN + F2X$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
F2X_filtered <- F2X %>% filter(total_REP1 > 49 & total_REP2 >49)
F2X_mean <- F2X_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
F2X_mean_ann <- F2X_filtered[ ,-c(2:23)]
F2X_mean_ann <- list(F2X_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
F2X_mean_ann$enzyme <- "F2X"
F2X_mean_ann$cell_line <- "MDA231"
# Split by editing type
F2X_any_editing <- F2X_mean_ann[,-c(2,3,4,5,6,8)]
names(F2X_any_editing)[2] <- "editing"
F2X_any_editing$editing_type <- "all"
F2X_tCT <- F2X_mean_ann[,-c(3,4,5,6,7,8)]
names(F2X_tCT)[2] <- "editing"
F2X_tCT$editing_type <- "C>T_only"
F2X_tCTN <- F2X_mean_ann[,-c(2,4,5,6,7,8)]
names(F2X_tCTN)[2] <- "editing"
F2X_tCTN$editing_type <- "C>T"
F2X_tCAN <- F2X_mean_ann[,-c(2,3,5,6,7,8)]
names(F2X_tCAN)[2] <- "editing"
F2X_tCAN$editing_type <- "C>A"
F2X_tCGN <- F2X_mean_ann[,-c(2,3,4,6,7,8)]
names(F2X_tCGN)[2] <- "editing"
F2X_tCGN$editing_type <- "C>G"
F2X_stack <- rbind(F2X_tCT,F2X_tCTN,F2X_tCAN,F2X_tCGN)
F2X_stack <- F2X_stack[complete.cases(F2X_stack), ]

# HF1
HF1$percent_tCT = (HF1$percent_tCT_REP1 + HF1$percent_tCT_REP2)/2
HF1$percent_tCTN = (HF1$percent_tCTN_REP1 + HF1$percent_tCTN_REP2)/2
HF1$percent_tCAN = (HF1$percent_tCAN_REP1 + HF1$percent_tCAN_REP2)/2
HF1$percent_tCGN = (HF1$percent_tCGN_REP1 + HF1$percent_tCGN_REP2)/2
HF1$percent_indel = (HF1$percent_indel_REP1 + HF1$percent_indel_REP2)/2
HF1$percent_editing = (HF1$percent_tCTN + HF1$percent_tCGN + HF1$percent_tCAN + HF1$percent_indel)
HF1$percent_BE = (HF1$percent_tCTN + HF1$percent_tCGN + HF1$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
HF1_filtered <- HF1 %>% filter(total_REP1 > 49 & total_REP2 >49)
HF1_mean <- HF1_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
HF1_mean_ann <- HF1_filtered[ ,-c(2:23)]
HF1_mean_ann <- list(HF1_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
HF1_mean_ann$enzyme <- "HF1"
HF1_mean_ann$cell_line <- "MDA231"
# Split by editing type
HF1_any_editing <- HF1_mean_ann[,-c(2,3,4,5,6,8)]
names(HF1_any_editing)[2] <- "editing"
HF1_any_editing$editing_type <- "all"
HF1_tCT <- HF1_mean_ann[,-c(3,4,5,6,7,8)]
names(HF1_tCT)[2] <- "editing"
HF1_tCT$editing_type <- "C>T_only"
HF1_tCTN <- HF1_mean_ann[,-c(2,4,5,6,7,8)]
names(HF1_tCTN)[2] <- "editing"
HF1_tCTN$editing_type <- "C>T"
HF1_tCAN <- HF1_mean_ann[,-c(2,3,5,6,7,8)]
names(HF1_tCAN)[2] <- "editing"
HF1_tCAN$editing_type <- "C>A"
HF1_tCGN <- HF1_mean_ann[,-c(2,3,4,6,7,8)]
names(HF1_tCGN)[2] <- "editing"
HF1_tCGN$editing_type <- "C>G"
HF1_stack <- rbind(HF1_tCT,HF1_tCTN,HF1_tCAN,HF1_tCGN)
HF1_stack <- HF1_stack[complete.cases(HF1_stack), ]

# HiFi
HiFi$percent_tCT = (HiFi$percent_tCT_REP1 + HiFi$percent_tCT_REP2)/2
HiFi$percent_tCTN = (HiFi$percent_tCTN_REP1 + HiFi$percent_tCTN_REP2)/2
HiFi$percent_tCAN = (HiFi$percent_tCAN_REP1 + HiFi$percent_tCAN_REP2)/2
HiFi$percent_tCGN = (HiFi$percent_tCGN_REP1 + HiFi$percent_tCGN_REP2)/2
HiFi$percent_indel = (HiFi$percent_indel_REP1 + HiFi$percent_indel_REP2)/2
HiFi$percent_editing = (HiFi$percent_tCTN + HiFi$percent_tCGN + HiFi$percent_tCAN + HiFi$percent_indel)
HiFi$percent_BE = (HiFi$percent_tCTN + HiFi$percent_tCGN + HiFi$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
HiFi_filtered <- HiFi %>% filter(total_REP1 > 49 & total_REP2 >49)
HiFi_mean <- HiFi_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
HiFi_mean_ann <- HiFi_filtered[ ,-c(2:23)]
HiFi_mean_ann <- list(HiFi_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
HiFi_mean_ann$enzyme <- "HiFi"
HiFi_mean_ann$cell_line <- "MDA231"
# Split by editing type
HiFi_any_editing <- HiFi_mean_ann[,-c(2,3,4,5,6,8)]
names(HiFi_any_editing)[2] <- "editing"
HiFi_any_editing$editing_type <- "all"
HiFi_tCT <- HiFi_mean_ann[,-c(3,4,5,6,7,8)]
names(HiFi_tCT)[2] <- "editing"
HiFi_tCT$editing_type <- "C>T_only"
HiFi_tCTN <- HiFi_mean_ann[,-c(2,4,5,6,7,8)]
names(HiFi_tCTN)[2] <- "editing"
HiFi_tCTN$editing_type <- "C>T"
HiFi_tCAN <- HiFi_mean_ann[,-c(2,3,5,6,7,8)]
names(HiFi_tCAN)[2] <- "editing"
HiFi_tCAN$editing_type <- "C>A"
HiFi_tCGN <- HiFi_mean_ann[,-c(2,3,4,6,7,8)]
names(HiFi_tCGN)[2] <- "editing"
HiFi_tCGN$editing_type <- "C>G"
HiFi_stack <- rbind(HiFi_tCT,HiFi_tCTN,HiFi_tCAN,HiFi_tCGN)
HiFi_stack <- HiFi_stack[complete.cases(HiFi_stack), ]

# HiFiNG
HiFiNG$percent_tCT = (HiFiNG$percent_tCT_REP1 + HiFiNG$percent_tCT_REP2)/2
HiFiNG$percent_tCTN = (HiFiNG$percent_tCTN_REP1 + HiFiNG$percent_tCTN_REP2)/2
HiFiNG$percent_tCAN = (HiFiNG$percent_tCAN_REP1 + HiFiNG$percent_tCAN_REP2)/2
HiFiNG$percent_tCGN = (HiFiNG$percent_tCGN_REP1 + HiFiNG$percent_tCGN_REP2)/2
HiFiNG$percent_indel = (HiFiNG$percent_indel_REP1 + HiFiNG$percent_indel_REP2)/2
HiFiNG$percent_editing = (HiFiNG$percent_tCTN + HiFiNG$percent_tCGN + HiFiNG$percent_tCAN + HiFiNG$percent_indel)
HiFiNG$percent_BE = (HiFiNG$percent_tCTN + HiFiNG$percent_tCGN + HiFiNG$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
HiFiNG_filtered <- HiFiNG %>% filter(total_REP1 > 49 & total_REP2 >49)
HiFiNG_mean <- HiFiNG_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
HiFiNG_mean_ann <- HiFiNG_filtered[ ,-c(2:23)]
HiFiNG_mean_ann <- list(HiFiNG_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
HiFiNG_mean_ann$enzyme <- "HiFiNG"
HiFiNG_mean_ann$cell_line <- "MDA231"
# Split by editing type
HiFiNG_any_editing <- HiFiNG_mean_ann[,-c(2,3,4,5,6,8)]
names(HiFiNG_any_editing)[2] <- "editing"
HiFiNG_any_editing$editing_type <- "all"
HiFiNG_tCT <- HiFiNG_mean_ann[,-c(3,4,5,6,7,8)]
names(HiFiNG_tCT)[2] <- "editing"
HiFiNG_tCT$editing_type <- "C>T_only"
HiFiNG_tCTN <- HiFiNG_mean_ann[,-c(2,4,5,6,7,8)]
names(HiFiNG_tCTN)[2] <- "editing"
HiFiNG_tCTN$editing_type <- "C>T"
HiFiNG_tCAN <- HiFiNG_mean_ann[,-c(2,3,5,6,7,8)]
names(HiFiNG_tCAN)[2] <- "editing"
HiFiNG_tCAN$editing_type <- "C>A"
HiFiNG_tCGN <- HiFiNG_mean_ann[,-c(2,3,4,6,7,8)]
names(HiFiNG_tCGN)[2] <- "editing"
HiFiNG_tCGN$editing_type <- "C>G"
HiFiNG_stack <- rbind(HiFiNG_tCT,HiFiNG_tCTN,HiFiNG_tCAN,HiFiNG_tCGN)
HiFiNG_stack <- HiFiNG_stack[complete.cases(HiFiNG_stack), ]

# FNLSNG
FNLSNG$percent_tCT = (FNLSNG$percent_tCT_REP1 + FNLSNG$percent_tCT_REP2)/2
FNLSNG$percent_tCTN = (FNLSNG$percent_tCTN_REP1 + FNLSNG$percent_tCTN_REP2)/2
FNLSNG$percent_tCAN = (FNLSNG$percent_tCAN_REP1 + FNLSNG$percent_tCAN_REP2)/2
FNLSNG$percent_tCGN = (FNLSNG$percent_tCGN_REP1 + FNLSNG$percent_tCGN_REP2)/2
FNLSNG$percent_indel = (FNLSNG$percent_indel_REP1 + FNLSNG$percent_indel_REP2)/2
FNLSNG$percent_editing = (FNLSNG$percent_tCTN + FNLSNG$percent_tCGN + FNLSNG$percent_tCAN + FNLSNG$percent_indel)
FNLSNG$percent_BE = (FNLSNG$percent_tCTN + FNLSNG$percent_tCGN + FNLSNG$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
FNLSNG_filtered <- FNLSNG %>% filter(total_REP1 > 49 & total_REP2 >49)
FNLSNG_mean <- FNLSNG_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
FNLSNG_mean_ann <- FNLSNG_filtered[ ,-c(2:23)]
FNLSNG_mean_ann <- list(FNLSNG_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
FNLSNG_mean_ann$enzyme <- "FNLSNG"
FNLSNG_mean_ann$cell_line <- "MDA231"
# Split by editing type
FNLSNG_any_editing <- FNLSNG_mean_ann[,-c(2,3,4,5,6,8)]
names(FNLSNG_any_editing)[2] <- "editing"
FNLSNG_any_editing$editing_type <- "all"
FNLSNG_tCT <- FNLSNG_mean_ann[,-c(3,4,5,6,7,8)]
names(FNLSNG_tCT)[2] <- "editing"
FNLSNG_tCT$editing_type <- "C>T_only"
FNLSNG_tCTN <- FNLSNG_mean_ann[,-c(2,4,5,6,7,8)]
names(FNLSNG_tCTN)[2] <- "editing"
FNLSNG_tCTN$editing_type <- "C>T"
FNLSNG_tCAN <- FNLSNG_mean_ann[,-c(2,3,5,6,7,8)]
names(FNLSNG_tCAN)[2] <- "editing"
FNLSNG_tCAN$editing_type <- "C>A"
FNLSNG_tCGN <- FNLSNG_mean_ann[,-c(2,3,4,6,7,8)]
names(FNLSNG_tCGN)[2] <- "editing"
FNLSNG_tCGN$editing_type <- "C>G"
FNLSNG_stack <- rbind(FNLSNG_tCT,FNLSNG_tCTN,FNLSNG_tCAN,FNLSNG_tCGN)
FNLSNG_stack <- FNLSNG_stack[complete.cases(FNLSNG_stack), ]

# FNLSVQR
FNLSVQR$percent_tCT = (FNLSVQR$percent_tCT_REP1 + FNLSVQR$percent_tCT_REP2)/2
FNLSVQR$percent_tCTN = (FNLSVQR$percent_tCTN_REP1 + FNLSVQR$percent_tCTN_REP2)/2
FNLSVQR$percent_tCAN = (FNLSVQR$percent_tCAN_REP1 + FNLSVQR$percent_tCAN_REP2)/2
FNLSVQR$percent_tCGN = (FNLSVQR$percent_tCGN_REP1 + FNLSVQR$percent_tCGN_REP2)/2
FNLSVQR$percent_indel = (FNLSVQR$percent_indel_REP1 + FNLSVQR$percent_indel_REP2)/2
FNLSVQR$percent_editing = (FNLSVQR$percent_tCTN + FNLSVQR$percent_tCGN + FNLSVQR$percent_tCAN + FNLSVQR$percent_indel)
FNLSVQR$percent_BE = (FNLSVQR$percent_tCTN + FNLSVQR$percent_tCGN + FNLSVQR$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
FNLSVQR_filtered <- FNLSVQR %>% filter(total_REP1 > 49 & total_REP2 >49)
FNLSVQR_mean <- FNLSVQR_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
FNLSVQR_mean_ann <- FNLSVQR_filtered[ ,-c(2:23)]
FNLSVQR_mean_ann <- list(FNLSVQR_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
FNLSVQR_mean_ann$enzyme <- "FNLSVQR"
FNLSVQR_mean_ann$cell_line <- "MDA231"
# Split by editing type
FNLSVQR_any_editing <- FNLSVQR_mean_ann[,-c(2,3,4,5,6,8)]
names(FNLSVQR_any_editing)[2] <- "editing"
FNLSVQR_any_editing$editing_type <- "all"
FNLSVQR_tCT <- FNLSVQR_mean_ann[,-c(3,4,5,6,7,8)]
names(FNLSVQR_tCT)[2] <- "editing"
FNLSVQR_tCT$editing_type <- "C>T_only"
FNLSVQR_tCTN <- FNLSVQR_mean_ann[,-c(2,4,5,6,7,8)]
names(FNLSVQR_tCTN)[2] <- "editing"
FNLSVQR_tCTN$editing_type <- "C>T"
FNLSVQR_tCAN <- FNLSVQR_mean_ann[,-c(2,3,5,6,7,8)]
names(FNLSVQR_tCAN)[2] <- "editing"
FNLSVQR_tCAN$editing_type <- "C>A"
FNLSVQR_tCGN <- FNLSVQR_mean_ann[,-c(2,3,4,6,7,8)]
names(FNLSVQR_tCGN)[2] <- "editing"
FNLSVQR_tCGN$editing_type <- "C>G"
FNLSVQR_stack <- rbind(FNLSVQR_tCT,FNLSVQR_tCTN,FNLSVQR_tCAN,FNLSVQR_tCGN)
FNLSVQR_stack <- FNLSVQR_stack[complete.cases(FNLSVQR_stack), ]

# xFNLS
xFNLS$percent_tCT = (xFNLS$percent_tCT_REP1 + xFNLS$percent_tCT_REP2)/2
xFNLS$percent_tCTN = (xFNLS$percent_tCTN_REP1 + xFNLS$percent_tCTN_REP2)/2
xFNLS$percent_tCAN = (xFNLS$percent_tCAN_REP1 + xFNLS$percent_tCAN_REP2)/2
xFNLS$percent_tCGN = (xFNLS$percent_tCGN_REP1 + xFNLS$percent_tCGN_REP2)/2
xFNLS$percent_indel = (xFNLS$percent_indel_REP1 + xFNLS$percent_indel_REP2)/2
xFNLS$percent_editing = (xFNLS$percent_tCTN + xFNLS$percent_tCGN + xFNLS$percent_tCAN + xFNLS$percent_indel)
xFNLS$percent_BE = (xFNLS$percent_tCTN + xFNLS$percent_tCGN + xFNLS$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
xFNLS_filtered <- xFNLS %>% filter(total_REP1 > 49 & total_REP2 >49)
xFNLS_mean <- xFNLS_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
xFNLS_mean_ann <- xFNLS_filtered[ ,-c(2:23)]
xFNLS_mean_ann <- list(xFNLS_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
xFNLS_mean_ann$enzyme <- "xFNLS"
xFNLS_mean_ann$cell_line <- "MDA231"
# Split by editing type
xFNLS_any_editing <- xFNLS_mean_ann[,-c(2,3,4,5,6,8)]
names(xFNLS_any_editing)[2] <- "editing"
xFNLS_any_editing$editing_type <- "all"
xFNLS_tCT <- xFNLS_mean_ann[,-c(3,4,5,6,7,8)]
names(xFNLS_tCT)[2] <- "editing"
xFNLS_tCT$editing_type <- "C>T_only"
xFNLS_tCTN <- xFNLS_mean_ann[,-c(2,4,5,6,7,8)]
names(xFNLS_tCTN)[2] <- "editing"
xFNLS_tCTN$editing_type <- "C>T"
xFNLS_tCAN <- xFNLS_mean_ann[,-c(2,3,5,6,7,8)]
names(xFNLS_tCAN)[2] <- "editing"
xFNLS_tCAN$editing_type <- "C>A"
xFNLS_tCGN <- xFNLS_mean_ann[,-c(2,3,4,6,7,8)]
names(xFNLS_tCGN)[2] <- "editing"
xFNLS_tCGN$editing_type <- "C>G"
xFNLS_stack <- rbind(xFNLS_tCT,xFNLS_tCTN,xFNLS_tCAN,xFNLS_tCGN)
xFNLS_stack <- xFNLS_stack[complete.cases(xFNLS_stack), ]

# scFNLS
scFNLS$percent_tCT = (scFNLS$percent_tCT_REP1 + scFNLS$percent_tCT_REP2)/2
scFNLS$percent_tCTN = (scFNLS$percent_tCTN_REP1 + scFNLS$percent_tCTN_REP2)/2
scFNLS$percent_tCAN = (scFNLS$percent_tCAN_REP1 + scFNLS$percent_tCAN_REP2)/2
scFNLS$percent_tCGN = (scFNLS$percent_tCGN_REP1 + scFNLS$percent_tCGN_REP2)/2
scFNLS$percent_indel = (scFNLS$percent_indel_REP1 + scFNLS$percent_indel_REP2)/2
scFNLS$percent_editing = (scFNLS$percent_tCTN + scFNLS$percent_tCGN + scFNLS$percent_tCAN + scFNLS$percent_indel)
scFNLS$percent_BE = (scFNLS$percent_tCTN + scFNLS$percent_tCGN + scFNLS$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
scFNLS_filtered <- scFNLS %>% filter(total_REP1 > 49 & total_REP2 >49)
scFNLS_mean <- scFNLS_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
scFNLS_mean_ann <- scFNLS_filtered[ ,-c(2:23)]
scFNLS_mean_ann <- list(scFNLS_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
scFNLS_mean_ann$enzyme <- "scFNLS"
scFNLS_mean_ann$cell_line <- "MDA231"
# Split by editing type
scFNLS_any_editing <- scFNLS_mean_ann[,-c(2,3,4,5,6,8)]
names(scFNLS_any_editing)[2] <- "editing"
scFNLS_any_editing$editing_type <- "all"
scFNLS_tCT <- scFNLS_mean_ann[,-c(3,4,5,6,7,8)]
names(scFNLS_tCT)[2] <- "editing"
scFNLS_tCT$editing_type <- "C>T_only"
scFNLS_tCTN <- scFNLS_mean_ann[,-c(2,4,5,6,7,8)]
names(scFNLS_tCTN)[2] <- "editing"
scFNLS_tCTN$editing_type <- "C>T"
scFNLS_tCAN <- scFNLS_mean_ann[,-c(2,3,5,6,7,8)]
names(scFNLS_tCAN)[2] <- "editing"
scFNLS_tCAN$editing_type <- "C>A"
scFNLS_tCGN <- scFNLS_mean_ann[,-c(2,3,4,6,7,8)]
names(scFNLS_tCGN)[2] <- "editing"
scFNLS_tCGN$editing_type <- "C>G"
scFNLS_stack <- rbind(scFNLS_tCT,scFNLS_tCTN,scFNLS_tCAN,scFNLS_tCGN)
scFNLS_stack <- scFNLS_stack[complete.cases(scFNLS_stack), ]

# BE4max
BE4max$percent_tCT = (BE4max$percent_tCT_REP1 + BE4max$percent_tCT_REP2)/2
BE4max$percent_tCTN = (BE4max$percent_tCTN_REP1 + BE4max$percent_tCTN_REP2)/2
BE4max$percent_tCAN = (BE4max$percent_tCAN_REP1 + BE4max$percent_tCAN_REP2)/2
BE4max$percent_tCGN = (BE4max$percent_tCGN_REP1 + BE4max$percent_tCGN_REP2)/2
BE4max$percent_indel = (BE4max$percent_indel_REP1 + BE4max$percent_indel_REP2)/2
BE4max$percent_editing = (BE4max$percent_tCTN + BE4max$percent_tCGN + BE4max$percent_tCAN + BE4max$percent_indel)
BE4max$percent_BE = (BE4max$percent_tCTN + BE4max$percent_tCGN + BE4max$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
BE4max_filtered <- BE4max %>% filter(total_REP1 > 49 & total_REP2 >49)
BE4max_mean <- BE4max_filtered[ ,-c(2:23)]
# Create seBE4maxate mean file, rename columns, add variable and enzyme annoation
BE4max_mean_ann <- BE4max_filtered[ ,-c(2:23)]
BE4max_mean_ann <- list(BE4max_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
BE4max_mean_ann$enzyme <- "BE4max"
BE4max_mean_ann$cell_line <- "MDA231"
# Split by editing type
BE4max_any_editing <- BE4max_mean_ann[,-c(2,3,4,5,6,8)]
names(BE4max_any_editing)[2] <- "editing"
BE4max_any_editing$editing_type <- "all"
BE4max_tCT <- BE4max_mean_ann[,-c(3,4,5,6,7,8)]
names(BE4max_tCT)[2] <- "editing"
BE4max_tCT$editing_type <- "C>T_only"
BE4max_tCTN <- BE4max_mean_ann[,-c(2,4,5,6,7,8)]
names(BE4max_tCTN)[2] <- "editing"
BE4max_tCTN$editing_type <- "C>T"
BE4max_tCAN <- BE4max_mean_ann[,-c(2,3,5,6,7,8)]
names(BE4max_tCAN)[2] <- "editing"
BE4max_tCAN$editing_type <- "C>A"
BE4max_tCGN <- BE4max_mean_ann[,-c(2,3,4,6,7,8)]
names(BE4max_tCGN)[2] <- "editing"
BE4max_tCGN$editing_type <- "C>G"
BE4max_stack <- rbind(BE4max_tCT,BE4max_tCTN,BE4max_tCAN,BE4max_tCGN)
BE4max_stack <- BE4max_stack[complete.cases(BE4max_stack), ]


# Par
Par$percent_tCT = (Par$percent_tCT_REP1 + Par$percent_tCT_REP2)/2
Par$percent_tCTN = (Par$percent_tCTN_REP1 + Par$percent_tCTN_REP2)/2
Par$percent_tCAN = (Par$percent_tCAN_REP1 + Par$percent_tCAN_REP2)/2
Par$percent_tCGN = (Par$percent_tCGN_REP1 + Par$percent_tCGN_REP2)/2
Par$percent_indel = (Par$percent_indel_REP1 + Par$percent_indel_REP2)/2
Par$percent_editing = (Par$percent_tCTN + Par$percent_tCGN + Par$percent_tCAN + Par$percent_indel)
Par$percent_BE = (Par$percent_tCTN + Par$percent_tCGN + Par$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
Par_filtered <- Par %>% filter(total_REP1 > 49 & total_REP2 >49)
Par_mean <- Par_filtered[ ,-c(2:23)]
# Create separate mean file, rename columns, add variable and enzyme annoation
Par_mean_ann <- Par_filtered[ ,-c(2:23)]
Par_mean_ann <- list(Par_mean_ann,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
Par_mean_ann$enzyme <- "Par"
Par_mean_ann$cell_line <- "MDA231"
# Split by editing type
Par_any_editing <- Par_mean_ann[,-c(2,3,4,5,6,8)]
names(Par_any_editing)[2] <- "editing"
Par_any_editing$editing_type <- "all"
Par_tCT <- Par_mean_ann[,-c(3,4,5,6,7,8)]
names(Par_tCT)[2] <- "editing"
Par_tCT$editing_type <- "C>T_only"
Par_tCTN <- Par_mean_ann[,-c(2,4,5,6,7,8)]
names(Par_tCTN)[2] <- "editing"
Par_tCTN$editing_type <- "C>T"
Par_tCAN <- Par_mean_ann[,-c(2,3,5,6,7,8)]
names(Par_tCAN)[2] <- "editing"
Par_tCAN$editing_type <- "C>A"
Par_tCGN <- Par_mean_ann[,-c(2,3,4,6,7,8)]
names(Par_tCGN)[2] <- "editing"
Par_tCGN$editing_type <- "C>G"
Par_stack <- rbind(Par_tCT,Par_tCTN,Par_tCAN,Par_tCGN)
Par_stack <- Par_stack[complete.cases(Par_stack), ]

# Split by editing type
Cas9NG_any_editing <- Cas9NG_mean_ann[,-c(2,3,4,5,6,8)]
names(Cas9NG_any_editing)[2] <- "editing"
Cas9NG_any_editing$editing_type <- "all"
Cas9NG_tCT <- Cas9NG_mean_ann[,-c(3,4,5,6,7,8)]
names(Cas9NG_tCT)[2] <- "editing"
Cas9NG_tCT$editing_type <- "C>T_only"
Cas9NG_tCTN <- Cas9NG_mean_ann[,-c(2,4,5,6,7,8)]
names(Cas9NG_tCTN)[2] <- "editing"
Cas9NG_tCTN$editing_type <- "C>T"
Cas9NG_tCAN <- Cas9NG_mean_ann[,-c(2,3,5,6,7,8)]
names(Cas9NG_tCAN)[2] <- "editing"
Cas9NG_tCAN$editing_type <- "C>A"
Cas9NG_tCGN <- Cas9NG_mean_ann[,-c(2,3,4,6,7,8)]
names(Cas9NG_tCGN)[2] <- "editing"
Cas9NG_tCGN$editing_type <- "C>G"
Cas9NG_stack <- rbind(Cas9NG_tCT,Cas9NG_tCTN,Cas9NG_tCAN,Cas9NG_tCGN)
Cas9NG_stack <- Cas9NG_stack[complete.cases(Cas9NG_stack), ]

```

Finally, join the data, calculate editing features like canonical (C>T) and non-canonical (C>A or C>G) ratios, and editing differences between different enzymes

```{r echo=TRUE, message=FALSE, warning=FALSE}
# Join mean data, add variables and remove cases with NAs
ALLPAM <- list(Cas9_mean,Cas9NG_mean,RA2_mean,BE4max_mean,F2X_mean,FNLSNG_mean,HF1_mean,HiFi_mean,HiFiNG_mean,FNLSVQR_mean,xFNLS_mean,scFNLS_mean,AP_variables) %>% 
  reduce(full_join, by = 'guide_ID')
ALLPAM <- ALLPAM[complete.cases(ALLPAM), ]
colnames(ALLPAM) <- c("guide_ID", "Cas9_percent_tCT", "Cas9_percent_tCTN", "Cas9_percent_tCAN", "Cas9_percent_tCGN",
                                    "Cas9_percent_indels", "Cas9_percent_editing", "Cas9_percent_BE",
                      "Cas9NG_percent_tCT", "Cas9NG_percent_tCTN", "Cas9NG_percent_tCAN", "Cas9NG_percent_tCGN",
                                    "Cas9NG_percent_indels", "Cas9NG_percent_editing", "Cas9NG_percent_BE",
                      "FNLS_percent_tCT", "FNLS_percent_tCTN", "FNLS_percent_tCAN", "FNLS_percent_tCGN",
                                    "FNLS_percent_indels", "FNLS_percent_editing", "FNLS_percent_BE",
                      "BE4max_percent_tCT", "BE4max_percent_tCTN", "BE4max_percent_tCAN", "BE4max_percent_tCGN",
                                    "BE4max_percent_indels", "BE4max_percent_editing", "BE4max_percent_BE",
                      "F2X_percent_tCT", "F2X_percent_tCTN", "F2X_percent_tCAN", "F2X_percent_tCGN",
                                    "F2X_percent_indels", "F2X_percent_editing", "F2X_percent_BE",
                      "FNLSNG_percent_tCT", "FNLSNG_percent_tCTN", "FNLSNG_percent_tCAN", "FNLSNG_percent_tCGN",
                                    "FNLSNG_percent_indels", "FNLSNG_percent_editing", "FNLSNG_percent_BE",
                      "HF1_percent_tCT", "HF1_percent_tCTN", "HF1_percent_tCAN", "HF1_percent_tCGN",
                                    "HF1_percent_indels", "HF1_percent_editing", "HF1_percent_BE",
                      "HiFi_percent_tCT", "HiFi_percent_tCTN", "HiFi_percent_tCAN", "HiFi_percent_tCGN",
                                    "HiFi_percent_indels", "HiFi_percent_editing", "HiFi_percent_BE",
                      "HiFiNG_percent_tCT", "HiFiNG_percent_tCTN", "HiFiNG_percent_tCAN", "HiFiNG_percent_tCGN",
                                    "HiFiNG_percent_indels", "HiFiNG_percent_editing", "HiFiNG_percent_BE",
                      "FNLSVQR_percent_tCT", "FNLSVQR_percent_tCTN", "FNLSVQR_percent_tCAN", "FNLSVQR_percent_tCGN",
                                    "FNLSVQR_percent_indels", "FNLSVQR_percent_editing", "FNLSVQR_percent_BE",
                      "xFNLS_percent_tCT", "xFNLS_percent_tCTN", "xFNLS_percent_tCAN", "xFNLS_percent_tCGN",
                                    "xFNLS_percent_indels", "xFNLS_percent_editing", "xFNLS_percent_BE",
                      "scFNLS_percent_tCT", "scFNLS_percent_tCTN", "scFNLS_percent_tCAN", "scFNLS_percent_tCGN",
                                    "scFNLS_percent_indels", "scFNLS_percent_editing", "scFNLS_percent_BE",
                    "symbol", "mutation", "guide_group","PAM", "PAM_class", "species",
                    "target","PAM_plus","exPAM")


# Rbind dataframes and remove rows with NAs
ALLPAM_stack <- rbind(Par_mean_ann,Cas9_mean_ann,Cas9NG_mean_ann,RA2_mean_ann,BE4max_mean_ann,F2X_mean_ann,
                               FNLSNG_mean_ann,HF1_mean_ann,HiFi_mean_ann,HiFiNG_mean_ann,
                               FNLSVQR_mean_ann,xFNLS_mean_ann,scFNLS_mean_ann)
# Remove poorly represented mouse p53 guides
ALLPAM_stack_filtered <- ALLPAM_stack[!grepl("Mouse_P53", ALLPAM_stack$guide_ID),]

```

Write the table to a .csv file 

```{r echo=TRUE, message=FALSE, warning=FALSE}
write_csv(ALLPAM, "~/Box Sync/BE_sensor/MDA231_allPAM/results/ALLPAM_MDA231.csv")
```

<p style="font-family: Arial; font-size:18pt; font-style:bold">
    Plot the data
</p>
Calculate and plot Pearson correlation coefficients for target cytosine editing within each replicate. Save .png files to working directory

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=16,fig.height=4}
# Cas9
a <- ggscatter(Cas9_filtered, x = "percent_indel_REP1", y = "percent_indel_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP Cas9 replicates",
          xlab = "Cas9 % Indels (REP1)", 
          ylab = "Cas9 % Indels (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_AP_Cas9_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# Cas9NG
b <- ggscatter(Cas9NG_filtered, x = "percent_indel_REP1", y = "percent_indel_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", fill = "grey50",
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP Cas9-NG replicates",
          xlab = "Cas9-NG % Indels (REP1)",
          ylab = "Cas9-NG % Indels (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_AP_Cas9NG_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# FNLS
c <- ggscatter(RA2_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP FNLS replicates",
          xlab = "FNLS %C>T editing (REP1)", 
          ylab = "FNLS %C>T editing (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_AP_FNLS_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# AncBE4max
d <- ggscatter(BE4max_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP AncBE4max replicates",
          xlab = "AncBE4max %C>T editing (REP1)", 
          ylab = "AncBE4max %C>T editing (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_AP_FNLS_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# F2X
e <- ggscatter(F2X_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP F2X replicates",
          xlab = "F2X %C>T editing (REP1)", 
          ylab = "F2X %C>T editing (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_F2X_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# FNLSNG
f <- ggscatter(FNLSNG_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP FNLS-NG replicates",
          xlab = "FNLS-NG %C>T editing (REP1)", 
          ylab = "FNLS-NG %C>T editing (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_FNLSNG_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# HF1
g <- ggscatter(HF1_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP FNLS-HF1 replicates",
          xlab = "FNLS-HF1 %C>T editing (REP1)", 
          ylab = "FNLS-HF1 %C>T editing (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_FNLS_HF1_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# HiFi
h <- ggscatter(HiFi_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP FNLS-HiFi replicates",
          xlab = "FNLS-HiFi %C>T editing (REP1)", 
          ylab = "FNLS-HiFi %C>T editing (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_FNLS_HiFi_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# HiFiNG
i <- ggscatter(HiFiNG_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP FNLS-HiFi-NG replicates",
          xlab = "FNLS-HiFi-NG %C>T editing (REP1)", 
          ylab = "FNLS-HiFi-NG %C>T editing (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_FNLS_HiFiNG_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# FNLSVQR
j <- ggscatter(FNLSVQR_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP FNLS-VQR replicates",
          xlab = "FNLS-VQR %C>T editing (REP1)", 
          ylab = "FNLS-VQR %C>T editing (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_FNLS_VQR_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# xFNLS
k <- ggscatter(xFNLS_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "AP xFNLS replicates",
          xlab = "xFNLS %C>T editing (REP1)", 
          ylab = "xFNLS %C>T editing (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_xFNLS_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# scFNLS
l <- ggscatter(scFNLS_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: AP scFNLS replicates",
          xlab = "scFNLS %C>T editing (REP1)", 
          ylab = "scFNLS %C>T editing (REP2)")+
  theme_bw(base_size = 12)+
  xlim(0,100)+
  ylim(0,100)+
  ggsave("MDA231_scFNLS_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")

plot_grid(a,b,c,d,e,f,g,h,i,j,k,l, labels = c('A','B','C','D','E','F','G','H','I','J','K','L'), ncol = 4, nrow = 3, label_size = 12)
```


Compare base editing and indel frequency

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=16,fig.height=4}
# RA2 (=FNLS)
ggscatter(ALLPAM_stack_filtered, x = "percent_BE", y = "percent_indel", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "PAM_class", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: FNLS BE vs indels",
          xlab = "Percent editing", 
          ylab = "Percent indels")+
  theme_bw()+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("ALLPAM_editing_vs_indels.png", dpi = 300, width = 10, height = 10, units = "cm")
```

Alternatively, you can plot the same data using `facet_grid` from the `HBES_mean_stack` dataframe

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=12,fig.height=3.5}
ggscatter(ALLPAM_stack_filtered, x = "percent_BE", y = "percent_indel", 
          alpha = 0.3, color = "PAM_class",
          title = "ALL PAM - MDA231",
          xlab = "Percent BE", 
          ylab = "Percent indels")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+
  facet_grid(cols = vars(enzyme))+
  ggsave("ALLPAM_BE_vs_indels_byenzyme.png", 
         dpi = 300, width = 10, height = 10, units = "cm")
```

Compare enzymes for both target cytosine and ALL cytosine editing

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=10,fig.height=7.5}

ALLPAM_stack_filtered %>% mutate(enzyme = fct_relevel(enzyme, levels=c("plasmid","Par","CAS9","Cas9NG","RA2","BE4max","F2X","FNLSNG",
                                                                       "HF1","HiFi","HiFiNG","FNLSVQR","xFNLS","scFNLS"))) %>% 
                                   ggplot( aes(x = enzyme, y = percent_indel, color = PAM_class)) + 
                                   geom_jitter(width = 0.2, height = 0)+
  theme_bw(base_size = 10)

ALLPAM_stack_filtered %>% mutate(enzyme = fct_relevel(enzyme, levels=c("plasmid","Par","CAS9","Cas9NG","RA2","BE4max","F2X","FNLSNG",
                                                                       "HF1","HiFi","HiFiNG","FNLSVQR","xFNLS","scFNLS"))) %>% 
                                   ggplot( aes(x = enzyme, y = percent_BE, color = PAM_class)) + 
                                   geom_jitter(width = 0.2, height = 0)+
  theme_bw(base_size = 10)

  ggsave("MDA231_HBES_FNLSvsF2X_ALLediting.png", dpi = 300, width = 10, height = 10, units = "cm")



```


Non-canonical editing by dinucleotide context

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=8,fig.height=4.5}
# Boxplot with stats (editing efficiency)
# FNLS vs high fidelity variants
ALLPAM_stack[ALLPAM_stack$enzyme == c("RA2","HF1","HiFi"),] %>%
  ggboxplot(x = "enzyme", y = "percent_BE",
            color = "enzyme", 
            palette = "jco",
            alpha = 0.5,
            title = "BE efficiency",
            ylab = "BE %",
            xlab = "enzyme",
            ggtheme = theme_bw(base_size = 10))+
  facet_grid(cols = vars(PAM_class))+
  stat_compare_means(comparisons = list(c("RA2","HF1"),c("RA2","HiFi"),c("HF1","HiFi")))+
  rotate_x_text()
# No stats
ALLPAM_stack[ALLPAM_stack$enzyme == c("RA2","HF1","HiFi"),] %>%
  ggboxplot(x = "enzyme", y = "percent_BE",
            color = "enzyme", 
            palette = "jco",
            alpha = 0.5,
            title = "BE efficiency",
            ylab = "BE %",
            xlab = "enzyme",
            ggtheme = theme_bw(base_size = 10))+
  facet_grid(cols = vars(PAM_class))+
  rotate_x_text()

# NG vs HiFiNG
ALLPAM_stack[ALLPAM_stack$enzyme == c("FNLSNG","HiFiNG"),] %>%
  ggboxplot(x = "enzyme", y = "percent_BE",
            color = "enzyme", 
            palette = "jco",
            alpha = 0.5,
            title = "BE efficiency",
            ylab = "BE %",
            xlab = "enzyme",
            ggtheme = theme_bw(base_size = 10))+
  facet_grid(cols = vars(PAM_class))+
  stat_compare_means(comparisons = list(c("FNLSNG","HiFiNG")))+
  rotate_x_text()
# No stats
ALLPAM_stack[ALLPAM_stack$enzyme == c("FNLSNG","HiFiNG"),] %>%
  ggboxplot(x = "enzyme", y = "percent_BE",
            color = "enzyme", 
            palette = "jco",
            alpha = 0.5,
            title = "BE efficiency",
            ylab = "BE %",
            xlab = "enzyme",
            ggtheme = theme_bw(base_size = 10))+
  facet_grid(cols = vars(PAM_class))+
  rotate_x_text()

# FNLS vs BE4max vs F2X
ALLPAM_stack[ALLPAM_stack$enzyme == c("RA2","BE4max","F2X"),] %>%
  ggboxplot(x = "enzyme", y = "percent_BE",
            color = "enzyme", 
            palette = "jco",
            alpha = 0.5,
            title = "BE efficiency",
            ylab = "BE %",
            xlab = "enzyme",
            ggtheme = theme_bw(base_size = 10))+
  facet_grid(cols = vars(PAM_class))+
  stat_compare_means(comparisons = list(c("RA2","BE4max"),c("RA2","F2X"),c("BE4max","F2X")))+
  rotate_x_text()
# No stats
ALLPAM_stack[ALLPAM_stack$enzyme == c("RA2","BE4max","F2X"),] %>%
  ggboxplot(x = "enzyme", y = "percent_BE",
            color = "enzyme", 
            palette = "jco",
            alpha = 0.5,
            title = "BE efficiency",
            ylab = "BE %",
            xlab = "enzyme",
            ggtheme = theme_bw(base_size = 10))+
  facet_grid(cols = vars(PAM_class))+
  rotate_x_text()
```

Plot editing data by heatmap
```{r}
# Heatmap (geom_tile)
base_plot <- ALLPAM_stack_filtered %>%
  mutate(enzyme = fct_relevel(enzyme, levels=c("CAS9","Cas9NG","RA2","BE4max","F2X","HF1","HiFi",
                                               "FNLSNG","HiFiNG","FNLSVQR","scFNLS","xFNLS"))) %>%
  ggplot( aes(x = guide_group, y = reorder(PAM, desc(PAM_class)), 
              fill = percent_tCTN)) + 
  geom_tile(stat = "identity", color = "white") +
  theme(axis.text.x = element_text(size = 7, angle = 90),
        axis.text.y=element_text(size=4), 
        axis.title.y = element_blank(), 
        legend.position = "right", 
        legend.key.width = unit(0.25, "cm"))+
  scale_fill_gradient(low = "#FFFFFF",
                      high = "#012345", limits = c(0,80), na.value = "pink")
# Separate plot by enzyme 
base_plot + facet_grid(. ~ enzyme)+ theme(strip.text.x = element_text(size = 6), legend.position = "bottom")

# Heatmap of human only (geom_tile)
base_plot_human <- ALLPAM_stack_filtered[ALLPAM_stack_filtered$Species == "Human",] %>%
  mutate(enzyme = fct_relevel(enzyme, levels=c("CAS9","Cas9NG","RA2","BE4max","F2X","HF1","HiFi",
                                               "FNLSNG","HiFiNG","FNLSVQR","scFNLS","xFNLS"))) %>%
  ggplot( aes(x = guide_group, y = reorder(PAM, desc(PAM_class)), 
              fill = percent_tCTN)) + 
  geom_tile(stat = "identity", color = "white") +
  theme(axis.text.x = element_text(size = 7, angle = 90),
        axis.text.y=element_text(size=4), 
        axis.title.y = element_blank(), 
        legend.position = "right", 
        legend.key.width = unit(0.25, "cm"))+
  scale_fill_gradient(low = "#FFFFFF",
                      high = "#012345", limits = c(0,80), na.value = "pink")
# Separate plot by enzyme 
base_plot_human + facet_grid(. ~ enzyme)+ theme(strip.text.x = element_text(size = 6), legend.position = "bottom")

# Heatmap of mouse only (geom_tile)
base_plot_mouse <- ALLPAM_stack_filtered[ALLPAM_stack_filtered$Species == "Mouse",] %>%
  mutate(enzyme = fct_relevel(enzyme, levels=c("CAS9","Cas9NG","RA2","BE4max","F2X","HF1","HiFi",
                                               "FNLSNG","HiFiNG","FNLSVQR","scFNLS","xFNLS"))) %>%
  ggplot( aes(x = guide_group, y = reorder(PAM, desc(PAM_class)), 
              fill = percent_tCTN)) + 
  geom_tile(stat = "identity", color = "white") +
  theme(axis.text.x = element_text(size = 7, angle = 90),
        axis.text.y=element_text(size=4), 
        axis.title.y = element_blank(), 
        legend.position = "right", 
        legend.key.width = unit(0.25, "cm"))+
  scale_fill_gradient(low = "#FFFFFF",
                      high = "#012345", limits = c(0,80), na.value = "pink")
# Separate plot by enzyme 
base_plot_mouse + facet_grid(. ~ enzyme)+ theme(strip.text.x = element_text(size = 6), legend.position = "bottom")
```

Plot indel data by heatmap
```{r}
# Heatmap (geom_tile)
base_plot <- ALLPAM_stack_filtered %>%
  mutate(enzyme = fct_relevel(enzyme, levels=c("CAS9","Cas9NG","RA2","BE4max","F2X","HF1","HiFi",
                                               "FNLSNG","HiFiNG","FNLSVQR","scFNLS","xFNLS"))) %>%
  ggplot( aes(x = guide_group, y = reorder(PAM, desc(PAM_class)), 
              fill = percent_indel)) + 
  geom_tile(stat = "identity", color = "white") +
  theme(axis.text.x = element_text(size = 7, angle = 90),
        axis.text.y=element_text(size=4), 
        axis.title.y = element_blank(), 
        legend.position = "right", 
        legend.key.width = unit(0.25, "cm"))+
  scale_fill_gradient(low = "#FFFFFF",
                      high = "#012345", limits = c(0,100), na.value = "pink")
# Separate plot by enzyme 
base_plot + facet_grid(. ~ enzyme)+ theme(strip.text.x = element_text(size = 6), legend.position = "bottom")

# Heatmap of human only (geom_tile)
base_plot_human <- ALLPAM_stack_filtered[ALLPAM_stack_filtered$Species == "Human",] %>%
  mutate(enzyme = fct_relevel(enzyme, levels=c("CAS9","Cas9NG","RA2","BE4max","F2X","HF1","HiFi",
                                               "FNLSNG","HiFiNG","FNLSVQR","scFNLS","xFNLS"))) %>%
  ggplot( aes(x = guide_group, y = reorder(PAM, desc(PAM_class)), 
              fill = percent_indel)) + 
  geom_tile(stat = "identity", color = "white") +
  theme(axis.text.x = element_text(size = 7, angle = 90),
        axis.text.y=element_text(size=4), 
        axis.title.y = element_blank(), 
        legend.position = "right", 
        legend.key.width = unit(0.25, "cm"))+
  scale_fill_gradient(low = "#FFFFFF",
                      high = "#012345", limits = c(0,100), na.value = "pink")
# Separate plot by enzyme 
base_plot_human + facet_grid(. ~ enzyme)+ theme(strip.text.x = element_text(size = 6), legend.position = "bottom")

# Heatmap of mouse only (geom_tile)
base_plot_mouse <- ALLPAM_stack_filtered[ALLPAM_stack_filtered$Species == "Mouse",] %>%
  mutate(enzyme = fct_relevel(enzyme, levels=c("CAS9","Cas9NG","RA2","BE4max","F2X","HF1","HiFi",
                                               "FNLSNG","HiFiNG","FNLSVQR","scFNLS","xFNLS"))) %>%
  ggplot( aes(x = guide_group, y = reorder(PAM, desc(PAM_class)), 
              fill = percent_indel)) + 
  geom_tile(stat = "identity", color = "white") +
  theme(axis.text.x = element_text(size = 7, angle = 90),
        axis.text.y=element_text(size=4), 
        axis.title.y = element_blank(), 
        legend.position = "right", 
        legend.key.width = unit(0.25, "cm"))+
  scale_fill_gradient(low = "#FFFFFF",
                      high = "#012345", limits = c(0,100), na.value = "pink")
# Separate plot by enzyme 
base_plot_mouse + facet_grid(. ~ enzyme)+ theme(strip.text.x = element_text(size = 6), legend.position = "bottom")
```
