---
title: "BE Sensor analysis: MDA231/HBES"
author: "Luke Dow & Henri Schmidt"
date: "11/16/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This is the pipeline fo the analysis of base editing sensor count data generated by bowtie alignment. This approach uses multiple reference alignments that iteratively measure C>T, C>A, C>G modification of target cytosines. Indels are calculated by subtracting all reads of the expected length (guide-scaffold-5'flank-target-3'flank), from the number of total reads that match the guide-scaffold-5'flank. 

First, load the required packages:

```{r echo=TRUE, message=FALSE, warning=FALSE}
library(dplyr)
library(tidyverse)
library(ggplot2)
library(viridis)
library(ggpubr)
library(cowplot)
library(gganimate)
library(readxl)
library(purrr)
library(stringr)

```

<p style="font-family: Arial; font-size:18pt; font-style:bold">
    Import transcript data
</p>
Import each of the tables containing alignment counts for each reference. The easiest way to do this is to set the working directory to the folder containing the relevant .txt files and import and rename columns using a loop:


```{r echo=TRUE, message=FALSE, warning=FALSE}
setwd("~/Box Sync/BE_sensor/MDA231/results/")
# Create list of file names
filenames <- gsub("\\.csv$","", list.files(pattern="\\.csv$"))
# Run import for loop
for(i in filenames){
  assign(i, read.csv(paste(i, ".csv", sep="")))
}
```

Next, import the metadata ("variables") associated with each of the guide_IDs

```{r echo=TRUE, message=FALSE, warning=FALSE}
HBES_variables <- read_excel("~/Box Sync/BE_sensor/Library_cloning/V4_v5/HBESv4_revised_whitelist.xlsx", sheet = "variables")
HBES_variables <- HBES_variables[,c(1:14)]
```

<p style="font-family: Arial; font-size:18pt; font-style:bold">
    Process the count data
</p>
The raw count data for each replicate is merged into a single dataframe and normalized to the total read counts for each guide by calculating "precent editing" for each type of editing event. The mean of each replicate is calculated. Where needed, some duplicate guides are removed as they cannot be mapped accurately.

```{r echo=TRUE, message=FALSE, warning=FALSE}
# Calculate mean percentages for TARGET data
MDA231_HBES_CAS9$percent_tCT = (MDA231_HBES_CAS9$percent_tCT_REP1 + MDA231_HBES_CAS9$percent_tCT_REP2)/2
MDA231_HBES_CAS9$percent_tCTN = (MDA231_HBES_CAS9$percent_tCTN_REP1 + MDA231_HBES_CAS9$percent_tCTN_REP2)/2
MDA231_HBES_CAS9$percent_tCAN = (MDA231_HBES_CAS9$percent_tCAN_REP1 + MDA231_HBES_CAS9$percent_tCAN_REP2)/2
MDA231_HBES_CAS9$percent_tCGN = (MDA231_HBES_CAS9$percent_tCGN_REP1 + MDA231_HBES_CAS9$percent_tCGN_REP2)/2
MDA231_HBES_CAS9$percent_indel = (MDA231_HBES_CAS9$percent_indel_REP1 + MDA231_HBES_CAS9$percent_indel_REP2)/2
MDA231_HBES_CAS9$percent_editing = (MDA231_HBES_CAS9$percent_tCTN + MDA231_HBES_CAS9$percent_tCGN +
                                          MDA231_HBES_CAS9$percent_tCAN + MDA231_HBES_CAS9$percent_indel)
MDA231_HBES_CAS9$percent_BE = (MDA231_HBES_CAS9$percent_tCTN + MDA231_HBES_CAS9$percent_tCGN +
                                          MDA231_HBES_CAS9$percent_tCAN)
# Calculate mean percentages for ALL data
MDA231_HBES_CAS9_ALL$percent_tCTN = (MDA231_HBES_CAS9_ALL$percent_tCTN_REP1 + MDA231_HBES_CAS9_ALL$percent_tCTN_REP2)/2
MDA231_HBES_CAS9_ALL$percent_tCAN = (MDA231_HBES_CAS9_ALL$percent_tCAN_REP1 + MDA231_HBES_CAS9_ALL$percent_tCAN_REP2)/2
MDA231_HBES_CAS9_ALL$percent_tCGN = (MDA231_HBES_CAS9_ALL$percent_tCGN_REP1 + MDA231_HBES_CAS9_ALL$percent_tCGN_REP2)/2
MDA231_HBES_CAS9_ALL$percent_BE = (MDA231_HBES_CAS9_ALL$percent_tCTN + MDA231_HBES_CAS9_ALL$percent_tCGN +
                                          MDA231_HBES_CAS9_ALL$percent_tCAN)
```

<p style="font-family: Arial; font-size:18pt; font-style:bold">
    Filter and merge the data
</p>
Data tables are filtered to retain only those guides that contain at least 100 reads in each of the replicates. Metadata ("variables") can be appended to the tables for later plotting purposes

```{r echo=TRUE, message=FALSE, warning=FALSE}
# Filter to 100+ reads/guide and remove replicate data
HBES_CAS9_filtered <- MDA231_HBES_CAS9 %>% filter(total_REP1 > 99 & total_REP2 >99)
HBES_CAS9_ALL_filtered <- MDA231_HBES_CAS9_ALL %>% filter(total_REP1 > 99 & total_REP2 >99)
HBES_CAS9_mean <- HBES_CAS9_filtered[ ,-c(2:23)]
HBES_CAS9_ALL_mean <- HBES_CAS9_ALL_filtered[ ,-c(5:22)]
HBES_CAS9_ALL_mean <- HBES_CAS9_ALL_mean[ ,c(1,5:8,2,3,4)]
# Create separate mean file, rename columns, add variable and enzyme annoation
HBES_CAS9_mean_ann <- HBES_CAS9_filtered[ ,-c(2:23)]
HBES_CAS9_mean_ann <- list(HBES_CAS9_mean_ann,HBES_variables) %>% 
  reduce(full_join, by = 'guide_ID')
HBES_CAS9_mean_ann$enzyme <- "CAS9"
HBES_CAS9_ALL_mean$enzyme <- "CAS9"
HBES_CAS9_mean_ann$cell_line <- "MDA231"
HBES_CAS9_ALL_mean$cell_line <- "MDA231"
# Split by editing type
HBES_CAS9_any_editing <- HBES_CAS9_mean_ann[,-c(2,3,4,5,6,8)]
names(HBES_CAS9_any_editing)[2] <- "editing"
HBES_CAS9_any_editing$editing_type <- "all"
HBES_CAS9_tCT <- HBES_CAS9_mean_ann[,-c(3,4,5,6,7,8)]
names(HBES_CAS9_tCT)[2] <- "editing"
HBES_CAS9_tCT$editing_type <- "C>T_only"
HBES_CAS9_tCTN <- HBES_CAS9_mean_ann[,-c(2,4,5,6,7,8)]
names(HBES_CAS9_tCTN)[2] <- "editing"
HBES_CAS9_tCTN$editing_type <- "C>T"
HBES_CAS9_tCAN <- HBES_CAS9_mean_ann[,-c(2,3,5,6,7,8)]
names(HBES_CAS9_tCAN)[2] <- "editing"
HBES_CAS9_tCAN$editing_type <- "C>A"
HBES_CAS9_tCGN <- HBES_CAS9_mean_ann[,-c(2,3,4,6,7,8)]
names(HBES_CAS9_tCGN)[2] <- "editing"
HBES_CAS9_tCGN$editing_type <- "C>G"
HBES_CAS9_stack <- rbind(HBES_CAS9_tCT,HBES_CAS9_tCTN,HBES_CAS9_tCAN,HBES_CAS9_tCGN)
HBES_CAS9_stack <- HBES_CAS9_stack[complete.cases(HBES_CAS9_stack), ]

# Add PAM_class, nucleotide context and cytosine position information to ALL data
HBES_CAS9_ALL_mean$PAM_class <- ifelse(grepl('GG$', HBES_CAS9_ALL_mean$PAM), "NGG", 
                         ifelse(grepl('GT$', HBES_CAS9_ALL_mean$PAM), "NGT",
                                ifelse(grepl('GC$', HBES_CAS9_ALL_mean$PAM), "NGC",
                                       ifelse(grepl('GA$', HBES_CAS9_ALL_mean$PAM), "NGA",
                                              ifelse(grepl('AG$', HBES_CAS9_ALL_mean$PAM), "NAG",
                                                     ifelse(grepl('TG$', HBES_CAS9_ALL_mean$PAM), "NNG",
                                                            ifelse(grepl('CG$', HBES_CAS9_ALL_mean$PAM), "NNG",
                                                                   ifelse(grepl('^GA', HBES_CAS9_ALL_mean$PAM), "GAN", "other"))))))))
colnames(HBES_CAS9_ALL_mean)[8] <- "nucleotide_context"
HBES_CAS9_ALL_mean$dinucleotide_context <- substr(HBES_CAS9_ALL_mean$nucleotide_context, 2,3)
HBES_CAS9_ALL_mean$trinucleotide_context <- substr(HBES_CAS9_ALL_mean$nucleotide_context, 2,4)
HBES_CAS9_ALL_mean$adj_cytosine_position <- ifelse(HBES_CAS9_ALL_mean$cytosine_position <10, HBES_CAS9_ALL_mean$cytosine_position -10, HBES_CAS9_ALL_mean$cytosine_position -9)
HBES_CAS9_ALL_mean$guide_ID_temp <- as.factor(paste(HBES_CAS9_ALL_mean$guide_ID, HBES_CAS9_ALL_mean$cytosine_position, sep = "_"))
HBES_CAS9_ALL_mean$guide_ID <- NULL
HBES_CAS9_ALL_mean <- HBES_CAS9_ALL_mean[ ,c(14,1:13)]
colnames(HBES_CAS9_ALL_mean)[1] <- "guide_ID"
# Split by editing type
HBES_CAS9_ALL_BE <- HBES_CAS9_ALL_mean[,-c(2,3,4)]
names(HBES_CAS9_ALL_BE)[2] <- "editing"
HBES_CAS9_ALL_BE$editing_type <- "BE"
HBES_CAS9_ALL_tCTN <- HBES_CAS9_ALL_mean[,-c(3,4,5)]
names(HBES_CAS9_ALL_tCTN)[2] <- "editing"
HBES_CAS9_ALL_tCTN$editing_type <- "C>T"
HBES_CAS9_ALL_tCAN <- HBES_CAS9_ALL_mean[,-c(2,4,5)]
names(HBES_CAS9_ALL_tCAN)[2] <- "editing"
HBES_CAS9_ALL_tCAN$editing_type <- "C>A"
HBES_CAS9_ALL_tCGN <- HBES_CAS9_ALL_mean[,-c(2,3,5)]
names(HBES_CAS9_ALL_tCGN)[2] <- "editing"
HBES_CAS9_ALL_tCGN$editing_type <- "C>G"
HBES_CAS9_ALL_stack <- rbind(HBES_CAS9_ALL_tCTN,HBES_CAS9_ALL_tCAN,HBES_CAS9_ALL_tCGN)
HBES_CAS9_ALL_stack <- HBES_CAS9_ALL_stack[complete.cases(HBES_CAS9_ALL_stack), ]
```

Repeat for each of the conditions (enzymes) included in the analysis

```{r echo=FALSE, message=FALSE, warning=FALSE}
# RA2
# Calculate mean percentages for TARGET data
MDA231_HBES_RA2$percent_tCT = (MDA231_HBES_RA2$percent_tCT_REP1 + MDA231_HBES_RA2$percent_tCT_REP2)/2
MDA231_HBES_RA2$percent_tCTN = (MDA231_HBES_RA2$percent_tCTN_REP1 + MDA231_HBES_RA2$percent_tCTN_REP2)/2
MDA231_HBES_RA2$percent_tCAN = (MDA231_HBES_RA2$percent_tCAN_REP1 + MDA231_HBES_RA2$percent_tCAN_REP2)/2
MDA231_HBES_RA2$percent_tCGN = (MDA231_HBES_RA2$percent_tCGN_REP1 + MDA231_HBES_RA2$percent_tCGN_REP2)/2
MDA231_HBES_RA2$percent_indel = (MDA231_HBES_RA2$percent_indel_REP1 + MDA231_HBES_RA2$percent_indel_REP2)/2
MDA231_HBES_RA2$percent_editing = (MDA231_HBES_RA2$percent_tCTN + MDA231_HBES_RA2$percent_tCGN +
                                          MDA231_HBES_RA2$percent_tCAN + MDA231_HBES_RA2$percent_indel)
MDA231_HBES_RA2$percent_BE = (MDA231_HBES_RA2$percent_tCTN + MDA231_HBES_RA2$percent_tCGN +
                                          MDA231_HBES_RA2$percent_tCAN)
# Calculate mean percentages for ALL data
MDA231_HBES_RA2_ALL$percent_tCTN = (MDA231_HBES_RA2_ALL$percent_tCTN_REP1 + MDA231_HBES_RA2_ALL$percent_tCTN_REP2)/2
MDA231_HBES_RA2_ALL$percent_tCAN = (MDA231_HBES_RA2_ALL$percent_tCAN_REP1 + MDA231_HBES_RA2_ALL$percent_tCAN_REP2)/2
MDA231_HBES_RA2_ALL$percent_tCGN = (MDA231_HBES_RA2_ALL$percent_tCGN_REP1 + MDA231_HBES_RA2_ALL$percent_tCGN_REP2)/2
MDA231_HBES_RA2_ALL$percent_BE = (MDA231_HBES_RA2_ALL$percent_tCTN + MDA231_HBES_RA2_ALL$percent_tCGN +
                                          MDA231_HBES_RA2_ALL$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
HBES_RA2_filtered <- MDA231_HBES_RA2 %>% filter(total_REP1 > 99 & total_REP2 >99)
HBES_RA2_ALL_filtered <- MDA231_HBES_RA2_ALL %>% filter(total_REP1 > 99 & total_REP2 >99)
HBES_RA2_mean <- HBES_RA2_filtered[ ,-c(2:23)]
HBES_RA2_ALL_mean <- HBES_RA2_ALL_filtered[ ,-c(5:22)]
HBES_RA2_ALL_mean <- HBES_RA2_ALL_mean[ ,c(1,5:8,2,3,4)]
# Create separate mean file, rename columns, add variable and enzyme annoation
HBES_RA2_mean_ann <- HBES_RA2_filtered[ ,-c(2:23)]
HBES_RA2_mean_ann <- list(HBES_RA2_mean_ann,HBES_variables) %>% 
  reduce(full_join, by = 'guide_ID')
HBES_RA2_mean_ann$enzyme <- "FNLS"
HBES_RA2_ALL_mean$enzyme <- "FNLS"
HBES_RA2_mean_ann$cell_line <- "MDA231"
HBES_RA2_ALL_mean$cell_line <- "MDA231"
# Split by editing type
HBES_RA2_any_editing <- HBES_RA2_mean_ann[,-c(2,3,4,5,6,8)]
names(HBES_RA2_any_editing)[2] <- "editing"
HBES_RA2_any_editing$editing_type <- "all"
HBES_RA2_tCT <- HBES_RA2_mean_ann[,-c(3,4,5,6,7,8)]
names(HBES_RA2_tCT)[2] <- "editing"
HBES_RA2_tCT$editing_type <- "C>T_only"
HBES_RA2_tCTN <- HBES_RA2_mean_ann[,-c(2,4,5,6,7,8)]
names(HBES_RA2_tCTN)[2] <- "editing"
HBES_RA2_tCTN$editing_type <- "C>T"
HBES_RA2_tCAN <- HBES_RA2_mean_ann[,-c(2,3,5,6,7,8)]
names(HBES_RA2_tCAN)[2] <- "editing"
HBES_RA2_tCAN$editing_type <- "C>A"
HBES_RA2_tCGN <- HBES_RA2_mean_ann[,-c(2,3,4,6,7,8)]
names(HBES_RA2_tCGN)[2] <- "editing"
HBES_RA2_tCGN$editing_type <- "C>G"
HBES_RA2_stack <- rbind(HBES_RA2_tCT,HBES_RA2_tCTN,HBES_RA2_tCAN,HBES_RA2_tCGN)
HBES_RA2_stack <- HBES_RA2_stack[complete.cases(HBES_RA2_stack), ]

# Add PAM_class, nucleotide context and cytosine position information to ALL data
HBES_RA2_ALL_mean$PAM_class <- ifelse(grepl('GG$', HBES_RA2_ALL_mean$PAM), "NGG", 
                         ifelse(grepl('GT$', HBES_RA2_ALL_mean$PAM), "NGT",
                                ifelse(grepl('GC$', HBES_RA2_ALL_mean$PAM), "NGC",
                                       ifelse(grepl('GA$', HBES_RA2_ALL_mean$PAM), "NGA",
                                              ifelse(grepl('AG$', HBES_RA2_ALL_mean$PAM), "NAG",
                                                     ifelse(grepl('TG$', HBES_RA2_ALL_mean$PAM), "NNG",
                                                            ifelse(grepl('CG$', HBES_RA2_ALL_mean$PAM), "NNG",
                                                                   ifelse(grepl('^GA', HBES_RA2_ALL_mean$PAM), "GAN", "other"))))))))
colnames(HBES_RA2_ALL_mean)[8] <- "nucleotide_context"
HBES_RA2_ALL_mean$dinucleotide_context <- substr(HBES_RA2_ALL_mean$nucleotide_context, 2,3)
HBES_RA2_ALL_mean$trinucleotide_context <- substr(HBES_RA2_ALL_mean$nucleotide_context, 2,4)
HBES_RA2_ALL_mean$adj_cytosine_position <- ifelse(HBES_RA2_ALL_mean$cytosine_position <10, HBES_RA2_ALL_mean$cytosine_position -10, HBES_RA2_ALL_mean$cytosine_position -9)
HBES_RA2_ALL_mean$guide_ID_temp <- as.factor(paste(HBES_RA2_ALL_mean$guide_ID, HBES_RA2_ALL_mean$cytosine_position, sep = "_"))
HBES_RA2_ALL_mean$guide_ID <- NULL
HBES_RA2_ALL_mean <- HBES_RA2_ALL_mean[ ,c(14,1:13)]
colnames(HBES_RA2_ALL_mean)[1] <- "guide_ID"
# Split by editing type
HBES_RA2_ALL_BE <- HBES_RA2_ALL_mean[,-c(2,3,4)]
names(HBES_RA2_ALL_BE)[2] <- "editing"
HBES_RA2_ALL_BE$editing_type <- "BE"
HBES_RA2_ALL_tCTN <- HBES_RA2_ALL_mean[,-c(3,4,5)]
names(HBES_RA2_ALL_tCTN)[2] <- "editing"
HBES_RA2_ALL_tCTN$editing_type <- "C>T"
HBES_RA2_ALL_tCAN <- HBES_RA2_ALL_mean[,-c(2,4,5)]
names(HBES_RA2_ALL_tCAN)[2] <- "editing"
HBES_RA2_ALL_tCAN$editing_type <- "C>A"
HBES_RA2_ALL_tCGN <- HBES_RA2_ALL_mean[,-c(2,3,5)]
names(HBES_RA2_ALL_tCGN)[2] <- "editing"
HBES_RA2_ALL_tCGN$editing_type <- "C>G"
HBES_RA2_ALL_stack <- rbind(HBES_RA2_ALL_tCTN,HBES_RA2_ALL_tCAN,HBES_RA2_ALL_tCGN)
HBES_RA2_ALL_stack <- HBES_RA2_ALL_stack[complete.cases(HBES_RA2_ALL_stack), ]

# F2X
# Calculate mean percentages for TARGET data
MDA231_HBES_F2X$percent_tCT = (MDA231_HBES_F2X$percent_tCT_REP1 + MDA231_HBES_F2X$percent_tCT_REP2)/2
MDA231_HBES_F2X$percent_tCTN = (MDA231_HBES_F2X$percent_tCTN_REP1 + MDA231_HBES_F2X$percent_tCTN_REP2)/2
MDA231_HBES_F2X$percent_tCAN = (MDA231_HBES_F2X$percent_tCAN_REP1 + MDA231_HBES_F2X$percent_tCAN_REP2)/2
MDA231_HBES_F2X$percent_tCGN = (MDA231_HBES_F2X$percent_tCGN_REP1 + MDA231_HBES_F2X$percent_tCGN_REP2)/2
MDA231_HBES_F2X$percent_indel = (MDA231_HBES_F2X$percent_indel_REP1 + MDA231_HBES_F2X$percent_indel_REP2)/2
MDA231_HBES_F2X$percent_editing = (MDA231_HBES_F2X$percent_tCTN + MDA231_HBES_F2X$percent_tCGN +
                                          MDA231_HBES_F2X$percent_tCAN + MDA231_HBES_F2X$percent_indel)
MDA231_HBES_F2X$percent_BE = (MDA231_HBES_F2X$percent_tCTN + MDA231_HBES_F2X$percent_tCGN +
                                          MDA231_HBES_F2X$percent_tCAN)
# Calculate mean percentages for ALL data
MDA231_HBES_F2X_ALL$percent_tCTN = (MDA231_HBES_F2X_ALL$percent_tCTN_REP1 + MDA231_HBES_F2X_ALL$percent_tCTN_REP2)/2
MDA231_HBES_F2X_ALL$percent_tCAN = (MDA231_HBES_F2X_ALL$percent_tCAN_REP1 + MDA231_HBES_F2X_ALL$percent_tCAN_REP2)/2
MDA231_HBES_F2X_ALL$percent_tCGN = (MDA231_HBES_F2X_ALL$percent_tCGN_REP1 + MDA231_HBES_F2X_ALL$percent_tCGN_REP2)/2
MDA231_HBES_F2X_ALL$percent_BE = (MDA231_HBES_F2X_ALL$percent_tCTN + MDA231_HBES_F2X_ALL$percent_tCGN +
                                          MDA231_HBES_F2X_ALL$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
HBES_F2X_filtered <- MDA231_HBES_F2X %>% filter(total_REP1 > 99 & total_REP2 >99)
HBES_F2X_ALL_filtered <- MDA231_HBES_F2X_ALL %>% filter(total_REP1 > 99 & total_REP2 >99)
HBES_F2X_mean <- HBES_F2X_filtered[ ,-c(2:23)]
HBES_F2X_ALL_mean <- HBES_F2X_ALL_filtered[ ,-c(5:22)]
HBES_F2X_ALL_mean <- HBES_F2X_ALL_mean[ ,c(1,5:8,2,3,4)]
# Create separate mean file, rename columns, add variable and enzyme annoation
HBES_F2X_mean_ann <- HBES_F2X_filtered[ ,-c(2:23)]
HBES_F2X_mean_ann <- list(HBES_F2X_mean_ann,HBES_variables) %>% 
  reduce(full_join, by = 'guide_ID')
HBES_F2X_mean_ann$enzyme <- "F2X"
HBES_F2X_ALL_mean$enzyme <- "F2X"
HBES_F2X_mean_ann$cell_line <- "MDA231"
HBES_F2X_ALL_mean$cell_line <- "MDA231"
# Split by editing type
HBES_F2X_any_editing <- HBES_F2X_mean_ann[,-c(2,3,4,5,6,8)]
names(HBES_F2X_any_editing)[2] <- "editing"
HBES_F2X_any_editing$editing_type <- "all"
HBES_F2X_tCT <- HBES_F2X_mean_ann[,-c(3,4,5,6,7,8)]
names(HBES_F2X_tCT)[2] <- "editing"
HBES_F2X_tCT$editing_type <- "C>T_only"
HBES_F2X_tCTN <- HBES_F2X_mean_ann[,-c(2,4,5,6,7,8)]
names(HBES_F2X_tCTN)[2] <- "editing"
HBES_F2X_tCTN$editing_type <- "C>T"
HBES_F2X_tCAN <- HBES_F2X_mean_ann[,-c(2,3,5,6,7,8)]
names(HBES_F2X_tCAN)[2] <- "editing"
HBES_F2X_tCAN$editing_type <- "C>A"
HBES_F2X_tCGN <- HBES_F2X_mean_ann[,-c(2,3,4,6,7,8)]
names(HBES_F2X_tCGN)[2] <- "editing"
HBES_F2X_tCGN$editing_type <- "C>G"
HBES_F2X_stack <- rbind(HBES_F2X_tCT,HBES_F2X_tCTN,HBES_F2X_tCAN,HBES_F2X_tCGN)
HBES_F2X_stack <- HBES_F2X_stack[complete.cases(HBES_F2X_stack), ]

# Add PAM_class, nucleotide context and cytosine position information to ALL data
HBES_F2X_ALL_mean$PAM_class <- ifelse(grepl('GG$', HBES_F2X_ALL_mean$PAM), "NGG", 
                         ifelse(grepl('GT$', HBES_F2X_ALL_mean$PAM), "NGT",
                                ifelse(grepl('GC$', HBES_F2X_ALL_mean$PAM), "NGC",
                                       ifelse(grepl('GA$', HBES_F2X_ALL_mean$PAM), "NGA",
                                              ifelse(grepl('AG$', HBES_F2X_ALL_mean$PAM), "NAG",
                                                     ifelse(grepl('TG$', HBES_F2X_ALL_mean$PAM), "NNG",
                                                            ifelse(grepl('CG$', HBES_F2X_ALL_mean$PAM), "NNG",
                                                                   ifelse(grepl('^GA', HBES_F2X_ALL_mean$PAM), "GAN", "other"))))))))
colnames(HBES_F2X_ALL_mean)[8] <- "nucleotide_context"
HBES_F2X_ALL_mean$dinucleotide_context <- substr(HBES_F2X_ALL_mean$nucleotide_context, 2,3)
HBES_F2X_ALL_mean$trinucleotide_context <- substr(HBES_F2X_ALL_mean$nucleotide_context, 2,4)
HBES_F2X_ALL_mean$adj_cytosine_position <- ifelse(HBES_F2X_ALL_mean$cytosine_position <10, HBES_F2X_ALL_mean$cytosine_position -10, HBES_F2X_ALL_mean$cytosine_position -9)
HBES_F2X_ALL_mean$guide_ID_temp <- as.factor(paste(HBES_F2X_ALL_mean$guide_ID, HBES_F2X_ALL_mean$cytosine_position, sep = "_"))
HBES_F2X_ALL_mean$guide_ID <- NULL
HBES_F2X_ALL_mean <- HBES_F2X_ALL_mean[ ,c(14,1:13)]
colnames(HBES_F2X_ALL_mean)[1] <- "guide_ID"
# Split by editing type
HBES_F2X_ALL_BE <- HBES_F2X_ALL_mean[,-c(2,3,4)]
names(HBES_F2X_ALL_BE)[2] <- "editing"
HBES_F2X_ALL_BE$editing_type <- "BE"
HBES_F2X_ALL_tCTN <- HBES_F2X_ALL_mean[,-c(3,4,5)]
names(HBES_F2X_ALL_tCTN)[2] <- "editing"
HBES_F2X_ALL_tCTN$editing_type <- "C>T"
HBES_F2X_ALL_tCAN <- HBES_F2X_ALL_mean[,-c(2,4,5)]
names(HBES_F2X_ALL_tCAN)[2] <- "editing"
HBES_F2X_ALL_tCAN$editing_type <- "C>A"
HBES_F2X_ALL_tCGN <- HBES_F2X_ALL_mean[,-c(2,3,5)]
names(HBES_F2X_ALL_tCGN)[2] <- "editing"
HBES_F2X_ALL_tCGN$editing_type <- "C>G"
HBES_F2X_ALL_stack <- rbind(HBES_F2X_ALL_tCTN,HBES_F2X_ALL_tCAN,HBES_F2X_ALL_tCGN)
HBES_F2X_ALL_stack <- HBES_F2X_ALL_stack[complete.cases(HBES_F2X_ALL_stack), ]

# FNLSNG
# Calculate mean percentages for TARGET data
MDA231_HBES_FNLSNG$percent_tCT = (MDA231_HBES_FNLSNG$percent_tCT_REP1 + MDA231_HBES_FNLSNG$percent_tCT_REP2)/2
MDA231_HBES_FNLSNG$percent_tCTN = (MDA231_HBES_FNLSNG$percent_tCTN_REP1 + MDA231_HBES_FNLSNG$percent_tCTN_REP2)/2
MDA231_HBES_FNLSNG$percent_tCAN = (MDA231_HBES_FNLSNG$percent_tCAN_REP1 + MDA231_HBES_FNLSNG$percent_tCAN_REP2)/2
MDA231_HBES_FNLSNG$percent_tCGN = (MDA231_HBES_FNLSNG$percent_tCGN_REP1 + MDA231_HBES_FNLSNG$percent_tCGN_REP2)/2
MDA231_HBES_FNLSNG$percent_indel = (MDA231_HBES_FNLSNG$percent_indel_REP1 + MDA231_HBES_FNLSNG$percent_indel_REP2)/2
MDA231_HBES_FNLSNG$percent_editing = (MDA231_HBES_FNLSNG$percent_tCTN + MDA231_HBES_FNLSNG$percent_tCGN +
                                          MDA231_HBES_FNLSNG$percent_tCAN + MDA231_HBES_FNLSNG$percent_indel)
MDA231_HBES_FNLSNG$percent_BE = (MDA231_HBES_FNLSNG$percent_tCTN + MDA231_HBES_FNLSNG$percent_tCGN +
                                          MDA231_HBES_FNLSNG$percent_tCAN)
# Calculate mean percentages for ALL data
MDA231_HBES_FNLSNG_ALL$percent_tCTN = (MDA231_HBES_FNLSNG_ALL$percent_tCTN_REP1 + MDA231_HBES_FNLSNG_ALL$percent_tCTN_REP2)/2
MDA231_HBES_FNLSNG_ALL$percent_tCAN = (MDA231_HBES_FNLSNG_ALL$percent_tCAN_REP1 + MDA231_HBES_FNLSNG_ALL$percent_tCAN_REP2)/2
MDA231_HBES_FNLSNG_ALL$percent_tCGN = (MDA231_HBES_FNLSNG_ALL$percent_tCGN_REP1 + MDA231_HBES_FNLSNG_ALL$percent_tCGN_REP2)/2
MDA231_HBES_FNLSNG_ALL$percent_BE = (MDA231_HBES_FNLSNG_ALL$percent_tCTN + MDA231_HBES_FNLSNG_ALL$percent_tCGN +
                                          MDA231_HBES_FNLSNG_ALL$percent_tCAN)
# Filter to 100+ reads/guide and remove replicate data
HBES_FNLSNG_filtered <- MDA231_HBES_FNLSNG %>% filter(total_REP1 > 99 & total_REP2 >99)
HBES_FNLSNG_ALL_filtered <- MDA231_HBES_FNLSNG_ALL %>% filter(total_REP1 > 99 & total_REP2 >99)
HBES_FNLSNG_mean <- HBES_FNLSNG_filtered[ ,-c(2:23)]
HBES_FNLSNG_ALL_mean <- HBES_FNLSNG_ALL_filtered[ ,-c(5:22)]
HBES_FNLSNG_ALL_mean <- HBES_FNLSNG_ALL_mean[ ,c(1,5:8,2,3,4)]
# Create separate mean file, rename columns, add variable and enzyme annoation
HBES_FNLSNG_mean_ann <- HBES_FNLSNG_filtered[ ,-c(2:23)]
HBES_FNLSNG_mean_ann <- list(HBES_FNLSNG_mean_ann,HBES_variables) %>% 
  reduce(full_join, by = 'guide_ID')
HBES_FNLSNG_mean_ann$enzyme <- "FNLSNG"
HBES_FNLSNG_ALL_mean$enzyme <- "FNLSNG"
HBES_FNLSNG_mean_ann$cell_line <- "MDA231"
HBES_FNLSNG_ALL_mean$cell_line <- "MDA231"
# Split by editing type
HBES_FNLSNG_any_editing <- HBES_FNLSNG_mean_ann[,-c(2,3,4,5,6,8)]
names(HBES_FNLSNG_any_editing)[2] <- "editing"
HBES_FNLSNG_any_editing$editing_type <- "all"
HBES_FNLSNG_tCT <- HBES_FNLSNG_mean_ann[,-c(3,4,5,6,7,8)]
names(HBES_FNLSNG_tCT)[2] <- "editing"
HBES_FNLSNG_tCT$editing_type <- "C>T_only"
HBES_FNLSNG_tCTN <- HBES_FNLSNG_mean_ann[,-c(2,4,5,6,7,8)]
names(HBES_FNLSNG_tCTN)[2] <- "editing"
HBES_FNLSNG_tCTN$editing_type <- "C>T"
HBES_FNLSNG_tCAN <- HBES_FNLSNG_mean_ann[,-c(2,3,5,6,7,8)]
names(HBES_FNLSNG_tCAN)[2] <- "editing"
HBES_FNLSNG_tCAN$editing_type <- "C>A"
HBES_FNLSNG_tCGN <- HBES_FNLSNG_mean_ann[,-c(2,3,4,6,7,8)]
names(HBES_FNLSNG_tCGN)[2] <- "editing"
HBES_FNLSNG_tCGN$editing_type <- "C>G"
HBES_FNLSNG_stack <- rbind(HBES_FNLSNG_tCT,HBES_FNLSNG_tCTN,HBES_FNLSNG_tCAN,HBES_FNLSNG_tCGN)
HBES_FNLSNG_stack <- HBES_FNLSNG_stack[complete.cases(HBES_FNLSNG_stack), ]

# Add PAM_class, nucleotide context and cytosine position information to ALL data
HBES_FNLSNG_ALL_mean$PAM_class <- ifelse(grepl('GG$', HBES_FNLSNG_ALL_mean$PAM), "NGG", 
                         ifelse(grepl('GT$', HBES_FNLSNG_ALL_mean$PAM), "NGT",
                                ifelse(grepl('GC$', HBES_FNLSNG_ALL_mean$PAM), "NGC",
                                       ifelse(grepl('GA$', HBES_FNLSNG_ALL_mean$PAM), "NGA",
                                              ifelse(grepl('AG$', HBES_FNLSNG_ALL_mean$PAM), "NAG",
                                                     ifelse(grepl('TG$', HBES_FNLSNG_ALL_mean$PAM), "NNG",
                                                            ifelse(grepl('CG$', HBES_FNLSNG_ALL_mean$PAM), "NNG",
                                                                   ifelse(grepl('^GA', HBES_FNLSNG_ALL_mean$PAM), "GAN", "other"))))))))
colnames(HBES_FNLSNG_ALL_mean)[8] <- "nucleotide_context"
HBES_FNLSNG_ALL_mean$dinucleotide_context <- substr(HBES_FNLSNG_ALL_mean$nucleotide_context, 2,3)
HBES_FNLSNG_ALL_mean$trinucleotide_context <- substr(HBES_FNLSNG_ALL_mean$nucleotide_context, 2,4)
HBES_FNLSNG_ALL_mean$adj_cytosine_position <- ifelse(HBES_FNLSNG_ALL_mean$cytosine_position <10, HBES_FNLSNG_ALL_mean$cytosine_position -10, HBES_FNLSNG_ALL_mean$cytosine_position -9)
HBES_FNLSNG_ALL_mean$guide_ID_temp <- as.factor(paste(HBES_FNLSNG_ALL_mean$guide_ID, HBES_FNLSNG_ALL_mean$cytosine_position, sep = "_"))
HBES_FNLSNG_ALL_mean$guide_ID <- NULL
HBES_FNLSNG_ALL_mean <- HBES_FNLSNG_ALL_mean[ ,c(14,1:13)]
colnames(HBES_FNLSNG_ALL_mean)[1] <- "guide_ID"
# Split by editing type
HBES_FNLSNG_ALL_BE <- HBES_FNLSNG_ALL_mean[,-c(2,3,4)]
names(HBES_FNLSNG_ALL_BE)[2] <- "editing"
HBES_FNLSNG_ALL_BE$editing_type <- "BE"
HBES_FNLSNG_ALL_tCTN <- HBES_FNLSNG_ALL_mean[,-c(3,4,5)]
names(HBES_FNLSNG_ALL_tCTN)[2] <- "editing"
HBES_FNLSNG_ALL_tCTN$editing_type <- "C>T"
HBES_FNLSNG_ALL_tCAN <- HBES_FNLSNG_ALL_mean[,-c(2,4,5)]
names(HBES_FNLSNG_ALL_tCAN)[2] <- "editing"
HBES_FNLSNG_ALL_tCAN$editing_type <- "C>A"
HBES_FNLSNG_ALL_tCGN <- HBES_FNLSNG_ALL_mean[,-c(2,3,5)]
names(HBES_FNLSNG_ALL_tCGN)[2] <- "editing"
HBES_FNLSNG_ALL_tCGN$editing_type <- "C>G"
HBES_FNLSNG_ALL_stack <- rbind(HBES_FNLSNG_ALL_tCTN,HBES_FNLSNG_ALL_tCAN,HBES_FNLSNG_ALL_tCGN)
HBES_FNLSNG_ALL_stack <- HBES_FNLSNG_ALL_stack[complete.cases(HBES_FNLSNG_ALL_stack), ]
```

Finally, join the data, calculate editing features like canonical (C>T) and non-canonical (C>A or C>G) ratios, and editing differences between different enzymes

```{r echo=TRUE, message=FALSE, warning=FALSE}
# Join mean data, add variables and remove cases with NAs
HBES <- list(HBES_CAS9_mean,HBES_RA2_mean,HBES_F2X_mean,HBES_FNLSNG_mean,HBES_variables) %>% 
  reduce(full_join, by = 'guide_ID')
HBES <- HBES[complete.cases(HBES), ]
colnames(HBES) <- c("guide_ID", "Cas9_percent_tCT", "Cas9_percent_tCTN", "Cas9_percent_tCAN", "Cas9_percent_tCGN",
                                    "Cas9_percent_indels", "Cas9_percent_editing", "Cas9_percent_BE",
                                    "FNLS_percent_tCT", "FNLS_percent_tCTN", "FNLS_percent_tCAN", "FNLS_percent_tCGN",
                                    "FNLS_percent_indels", "FNLS_percent_editing", "FNLS_percent_BE",
                                    "F2X_percent_tCT", "F2X_percent_tCTN", "F2X_percent_tCAN", "F2X_percent_tCGN",
                                    "F2X_percent_indels", "F2X_percent_editing", "F2X_percent_BE",
                                    "FNLSNG_percent_tCT", "FNLSNG_percent_tCTN", "FNLSNG_percent_tCAN", "FNLSNG_percent_tCGN",
                                    "FNLSNG_percent_indels", "FNLSNG_percent_editing", "FNLSNG_percent_BE",
                    "protospacer", "PAM", "PAM_class","cytosines_in_window", "target_position",
                    "bi_nucleotide_context", "tri_nucleotide_context", "symbol", "variant","PAM_plus","exPAM")

HBES_ALL <- list(HBES_RA2_ALL_mean, HBES_F2X_ALL_mean, HBES_FNLSNG_ALL_mean) %>% reduce(full_join, by = 'guide_ID')
HBES_ALL <- HBES_ALL[complete.cases(HBES_ALL), ]
HBES_ALL <- HBES_ALL[ ,-c(6:14,19:27)]
colnames(HBES_ALL) <- c("guide_ID", "FNLS_percent_tCTN", "FNLS_percent_tCAN", "FNLS_percent_tCGN","FNLS_percent_BE",
                                    "F2X_percent_tCTN", "F2X_percent_tCAN", "F2X_percent_tCGN","F2X_percent_BE",
                                    "FNLSNG_percent_tCTN", "FNLSNG_percent_tCAN", "FNLSNG_percent_tCGN","FNLSNG_percent_BE",
                        "cytosine_position", "PAM","nucleotide_context","enzyme","cell_line","PAM_class","dinucleotide_context",
                        "trinucleotide_context", "adj_cytosine_position")

# Calculate editing ratios
HBES$F2X_FNLS_ratio <- (HBES$F2X_percent_tCTN / HBES$FNLS_percent_tCTN)
HBES$F2X_gain <- HBES$F2X_percent_tCTN - HBES$FNLS_percent_tCTN
HBES$NG_gain <- HBES$FNLSNG_percent_tCTN - HBES$FNLS_percent_tCTN
HBES_ALL$F2X_FNLS_ratio <- (HBES_ALL$F2X_percent_tCTN / HBES_ALL$FNLS_percent_tCTN)
HBES_ALL$F2X_gain <- HBES_ALL$F2X_percent_tCTN - HBES_ALL$FNLS_percent_tCTN
HBES_ALL$NG_gain <- HBES_ALL$FNLSNG_percent_tCTN - HBES_ALL$FNLS_percent_tCTN

# Rbind dataframes and remove rows with NAs
HBES_mean_stack <- rbind(HBES_CAS9_mean_ann,HBES_RA2_mean_ann,
                         HBES_F2X_mean_ann,HBES_FNLSNG_mean_ann)
HBES_mean_stack <- HBES_mean_stack[complete.cases(HBES_mean_stack), ]
HBES_stack_5percent_filter <- HBES_mean_stack %>% filter(percent_tCTN > 5)
HBES_stack_1percent_filter <- HBES_mean_stack %>% filter(percent_tCTN > 1)
HBES_stack_5percent_filter$GvsT_ratio = HBES_stack_5percent_filter$percent_tCGN/HBES_stack_5percent_filter$percent_tCTN
HBES_stack_5percent_filter$AvsT_ratio = HBES_stack_5percent_filter$percent_tCAN/HBES_stack_5percent_filter$percent_tCTN
HBES_stack_1percent_filter$HBES_MDA231_CTrank[order(-HBES_stack_1percent_filter$percent_tCTN)] <- 1:nrow(HBES_stack_1percent_filter)
HBES_stack_1percent_filter$HBES_MDA231_BErank[order(-HBES_stack_1percent_filter$percent_tCTN)] <- 1:nrow(HBES_stack_1percent_filter)
# ALL data
HBES_ALL_stack <- rbind(HBES_RA2_ALL_mean,
                         HBES_F2X_ALL_mean,HBES_FNLSNG_ALL_mean)
HBES_ALL_stack <- HBES_ALL_stack[complete.cases(HBES_ALL_stack), ]
HBES_ALL_stack_2percent_filter <- HBES_ALL_stack %>% filter(percent_tCTN > 2)
HBES_ALL_stack_2percent_filter$GvsT_ratio = HBES_ALL_stack_2percent_filter$percent_tCGN/HBES_ALL_stack_2percent_filter$percent_tCTN
HBES_ALL_stack_2percent_filter$AvsT_ratio = HBES_ALL_stack_2percent_filter$percent_tCAN/HBES_ALL_stack_2percent_filter$percent_tCTN

HBES_ALL_stack_editing_type <- rbind(HBES_RA2_ALL_stack,HBES_F2X_ALL_stack,HBES_FNLSNG_ALL_stack)

HBES_ALL_stack_5percent_filter <- HBES_ALL_stack %>% filter(percent_tCTN > 5)
HBES_ALL_stack_5percent_filter$GvsT_ratio = HBES_ALL_stack_5percent_filter$percent_tCGN/HBES_ALL_stack_5percent_filter$percent_tCTN
HBES_ALL_stack_5percent_filter$AvsT_ratio = HBES_ALL_stack_5percent_filter$percent_tCAN/HBES_ALL_stack_5percent_filter$percent_tCTN
# Calculate mean editing efficiencies across each cytosine position, separate by PAM, enzyme and dinucleotide context
HBES_percent_BE_mean <- aggregate(percent_BE ~ adj_cytosine_position+PAM_class+enzyme+dinucleotide_context, data = HBES_ALL_stack, mean)
HBES_percent_tCTN_mean <- aggregate(percent_tCTN ~ adj_cytosine_position+PAM_class+enzyme+dinucleotide_context, data = HBES_ALL_stack, mean)
HBES_percent_tCGN_mean <- aggregate(percent_tCGN ~ adj_cytosine_position+PAM_class+enzyme+dinucleotide_context, data = HBES_ALL_stack, mean)
HBES_percent_tCAN_mean <- aggregate(percent_tCAN ~ adj_cytosine_position+PAM_class+enzyme+dinucleotide_context, data = HBES_ALL_stack, mean)
```

Write the table to a .csv file 

```{r echo=TRUE, message=FALSE, warning=FALSE}
write_csv(HBES, "~/Box Sync/BE_sensor/MDA231/results/HBES_MDA231.csv")
write_csv(HBES_ALL, "~/Box Sync/BE_sensor/MDA231/results/HBES_ALL_MDA231.csv")
```

<p style="font-family: Arial; font-size:18pt; font-style:bold">
    Plot the data
</p>
Calculate and plot Pearson correlation coefficients for target cytosine editing within each replicate. Save .png files to working directory

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=16,fig.height=4}
a <- ggscatter(HBES_RA2_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: HBES FNLS replicates",
          xlab = "FNLS %C>T editing (REP1)", 
          ylab = "FNLS %C>T editing (REP2)")+
  theme_bw(base_size = 8)+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_RA2_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# F2X
b <- ggscatter(HBES_F2X_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", fill = "grey50",
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: HBES F2X replicates",
          xlab = "F2X %C>T editing (REP1)",
          ylab = "F2X %C>T editing (REP2)")+
  theme_bw(base_size = 8)+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_F2X_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# FNLS-NG
c <- ggscatter(HBES_FNLSNG_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: HBES FNLS-NG replicates",
          xlab = "FNLS-NG %C>T editing (REP1)", 
          ylab = "FNLS-NG %C>T editing (REP2)")+
  theme_bw(base_size = 8)+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_NG_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# CAS9
d <- ggscatter(HBES_CAS9_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: HBES CAS9 replicates",
          xlab = "FNLS-NG %C>T editing (REP1)", 
          ylab = "FNLS-NG %C>T editing (REP2)")+
  theme_bw(base_size = 8)+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_CAS9_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")

plot_grid(a,b,c,d, labels = c('A','B','C','D'), ncol = 4, label_size = 10)
```
```{r echo=FALSE, message=FALSE, warning=FALSE, fig.width=16,fig.height=4}
a <- ggscatter(HBES_RA2_ALL_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: HBES-ALL FNLS replicates",
          xlab = "FNLS %C>T editing (REP1)", 
          ylab = "FNLS % C>T editing (REP2)")+
  theme_bw(base_size = 8)+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_RA2_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# F2X
b <- ggscatter(HBES_F2X_ALL_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", fill = "grey50",
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: HBES-ALL F2X replicates",
          xlab = "F2X %C>T editing (REP1)",
          ylab = "F2X % C>T editing (REP2)")+
  theme_bw(base_size = 8)+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_F2X_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# FNLS-NG
c <- ggscatter(HBES_FNLSNG_ALL_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: HBES-ALL FNLS-NG replicates",
          xlab = "FNLS-NG %C>T editing (REP1)", 
          ylab = "FNLS-NG %C>T editing (REP2)")+
  theme_bw(base_size = 8)+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_NG_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")
# CAS9
d <- ggscatter(HBES_CAS9_ALL_filtered, x = "percent_tCTN_REP1", y = "percent_tCTN_REP2", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: HBES-ALL CAS9 replicates",
          xlab = "FNLS-NG %C>T editing (REP1)", 
          ylab = "FNLS-NG %C>T editing (REP2)")+
  theme_bw(base_size = 8)+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_CAS9_tCTN_replicates.png", dpi = 300, width = 10, height = 10, units = "cm")

plot_grid(a,b,c,d, labels = c('A','B','C','D'), ncol = 4, label_size = 10)
```

Compare base editing and indel frequency (only for target cytosine analysis)

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=16,fig.height=4}
# RA2 (=FNLS)
e <- ggscatter(HBES_RA2_filtered, x = "percent_BE", y = "percent_indel", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: FNLS BE vs indels",
          xlab = "Percent editing (FNLS)", 
          ylab = "Percent indels (FNLS)")+
  theme_bw()+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_RA2_editing_vs_indels.png", dpi = 300, width = 10, height = 10, units = "cm")
# F2X 
f <- ggscatter(HBES_F2X_filtered, x = "percent_BE", y = "percent_indel", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: F2X BE vs indels",
          xlab = "Percent editing (F2X)", 
          ylab = "Percent indels (F2X)")+
  theme_bw()+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_F2X_editing_vs_indels.png", dpi = 300, width = 10, height = 10, units = "cm")
g <- ggscatter(HBES_FNLSNG_filtered, x = "percent_BE", y = "percent_indel", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: FNLS-NG BE vs indels",
          xlab = "Percent editing (FNLS-NG)", 
          ylab = "Percent indels (FNLS-NG)")+
  theme_bw()+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_NG_editing_vs_indels.png", dpi = 300, width = 10, height = 10, units = "cm")
h <- ggscatter(HBES_CAS9_filtered, x = "percent_BE", y = "percent_indel", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: CAS9 BE vs indels",
          xlab = "Percent editing (FNLS-NG)", 
          ylab = "Percent indels (FNLS-NG)")+
  theme_bw()+
  xlim(0,80)+
  ylim(0,80)+
  ggsave("MDA231_HBES_CAS9_editing_vs_indels.png", dpi = 300, width = 10, height = 10, units = "cm")

plot_grid(e,f,g,h, labels = c('A','B','C','D'), ncol = 4, label_size = 10)
```

Alternatively, you can plot the same data using `facet_grid` from the `HBES_mean_stack` dataframe

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=12,fig.height=3.5}
ggscatter(HBES_mean_stack, x = "percent_BE", y = "percent_indel", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "steelblue4", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "Mouse library - MDA231",
          xlab = "Percent BE", 
          ylab = "Percent indels")+
  theme_bw(base_size = 10)+
  xlim(0,80)+
  ylim(0,80)+
  facet_grid(cols = vars(enzyme))+
  ggsave("MDA231_HBES_BE_vs_indels.png", 
         dpi = 300, width = 10, height = 10, units = "cm")
```

Compare enzymes for both target cytosine and ALL cytosine editing

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=10,fig.height=7.5}
i <- ggscatter(HBES, x = "FNLS_percent_editing", y = "F2X_percent_editing", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "FNLS_percent_tCTN", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: All editing - FNLS vs F2X",
          xlab = "FNLS % all editing", 
          ylab = "F2X % all editing")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+ 
  scale_color_viridis(option = "A")+
  ggsave("MDA231_HBES_FNLSvsF2X_ALLediting.png", dpi = 300, width = 10, height = 10, units = "cm")
# RA2 vs F2X (BE)
j <- ggscatter(HBES, x = "FNLS_percent_BE", y = "F2X_percent_BE", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "FNLS_percent_tCTN", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: base editing - FNLS vs F2X",
          xlab = "FNLS % base editing (C>T/A/G)", 
          ylab = "F2X % base editing (C>T/A/G)")+
  theme_bw(base_size = 10)+
  xlim(0,80)+
  ylim(0,80)+ 
  scale_color_viridis(option = "A")+
  ggsave("MDA231_HBES_FNLSvsF2X_ALLediting.png", dpi = 300, width = 10, height = 10, units = "cm")
# RA2 vs F2X (tCTN)
k <- ggscatter(HBES, x = "FNLS_percent_tCTN", y = "F2X_percent_tCTN", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "FNLS_percent_tCTN", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: target C>T editing - FNLS vs F2X",
          xlab = "FNLS % C>T editing (REP1)", 
          ylab = "F2X % C>T editing (REP2)")+
  theme_bw(base_size = 10)+
  xlim(0,80)+
  ylim(0,80)+ 
  scale_color_viridis(option = "A")+
  ggsave("MDA231_HBES_FNLSvsF2X_tCTN.png", dpi = 300, width = 10, height = 10, units = "cm")
# RA2 vs F2X (indel)
l <- ggscatter(HBES, x = "FNLS_percent_indels", y = "F2X_percent_indels", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "FNLS_percent_tCTN", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: indels - FNLS vs F2X",
          xlab = "FNLS indels", 
          ylab = "F2X indels")+
  theme_bw(base_size = 10)+
  xlim(0,80)+
  ylim(0,80)+ 
  scale_color_viridis(option = "A")+
  ggsave("MDA231_HBES_FNLSvsF2X_indel.png", dpi = 300, width = 10, height = 10, units = "cm")

m <- ggscatter(HBES, x = "Cas9_percent_indels", y = "F2X_percent_tCTN", alpha = 0.3, color = "PAM_class", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: indels - Cas9 indels vs F2X editing",
          xlab = "Cas9 indels", 
          ylab = "F2X C>T editing")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,80)+
  ggsave("MDA231_HBES_Cas9vsF2X_indelvsBE.png", dpi = 300, width = 10, height = 10, units = "cm")

n <- ggscatter(HBES[HBES$PAM_class == c("NGG","NAG","NGA"),], x = "Cas9_percent_indels", y = "F2X_percent_tCTN", alpha = 0.3, color = "PAM_class", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: indels - Cas9 indels vs F2X editing",
          xlab = "Cas9 indels", 
          ylab = "F2X C>T editing")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,80)+
  ggsave("MDA231_HBES_Cas9vsF2X_indelvsBE.png", dpi = 300, width = 10, height = 10, units = "cm")

plot_grid(i,j,k,l,m,n, labels = c('A','B','C','D','E','F'), ncol = 2, nrow = 3)
```

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=10,fig.height=3.5}
# RA2 vs F2X (BE)
j <- ggscatter(HBES_ALL, x = "FNLS_percent_BE", y = "F2X_percent_BE", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "adj_cytosine_position", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: ALL base editing - FNLS vs F2X",
          xlab = "FNLS % base editing (C>T/A/G)", 
          ylab = "F2X % base editing (C>T/A/G)")+
  theme_bw(base_size = 10)+
  xlim(0,100)+
  ylim(0,100)+ 
  scale_color_viridis(option = "A")+
  ggsave("MDA231_HBES_RA2vsF2X_ALLediting.png", dpi = 300, width = 10, height = 10, units = "cm")
# RA2 vs F2X (tCTN)
k <- ggscatter(HBES_ALL, x = "FNLS_percent_tCTN", y = "F2X_percent_tCTN", 
          add = "reg.line", conf.int = TRUE, alpha = 0.3, color = "adj_cytosine_position", 
          cor.coef = TRUE, cor.method = "pearson",
          title = "MDA231: ALL C>T editing - FNLS vs F2X",
          xlab = "FNLS % C>T editing", 
          ylab = "F2X % C>T editing")+
  theme_bw(base_size = 10)+
  xlim(0,80)+
  ylim(0,80)+ 
  scale_color_viridis(option = "A")+
  ggsave("MDA231_HBES_RA2vsF2X_CT.png", dpi = 300, width = 10, height = 10, units = "cm")

plot_grid(j,k, labels = c('A','B'), ncol = 2)
```

Compare base editing activity across target cytosine positions

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=5,fig.height=4}
ggplot(HBES)+ 
  geom_point( aes(x=FNLS_percent_tCTN, y=F2X_percent_tCTN, color = target_position), alpha = 0.5)+
  xlim(0,70)+ 
  ylim(0,70)+
  geom_abline(intercept = 0, slope = 1, color="grey25", linetype="dashed", size=0.65)+
  theme_bw(base_size = 10)+
  scale_color_viridis(option = "B")+
  labs(title = "FNLS vs F2X (HBES-MDA231)",
       x = "Percent target editing - FNLS",
       y = "Percent target editing - F2X")+
  ggsave("MDA231_HBES_RA2vsF2X_tCTN_correlation.png", dpi = 300, width = 10, height = 10, units = "cm")
```

Base editing range across target cytosine positions x dinucleotide context

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=6,fig.height=5}
# Compare types of editing for a single enzyme on same plot
# RA2
HBES_RA2_ALL_stack[HBES_RA2_ALL_stack$PAM_class == c("NGG"),] %>% 
  mutate(editing_type = fct_relevel(editing_type, levels=c("C>T","C>A","C>G"))) %>%
  ggplot( aes(x = adj_cytosine_position, y = editing, color = editing))+
  geom_jitter(size = .6, width = 0.1, alpha = 0.4)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))+
  ylim(0,80) + facet_grid(rows = vars(dinucleotide_context), cols = vars(editing_type))+
  theme_bw(base_size = 10)+
  labs(title = "FNLS editing - NGG ONLY",
       xlab = "Cytosine position",
       ylab = "Percent editing",
       color = NULL)+ 
  scale_color_viridis(option = "A", limits = c(0,80))
#F2X
HBES_F2X_ALL_stack[HBES_F2X_ALL_stack$PAM_class == c("NGG"),] %>% 
  mutate(editing_type = fct_relevel(editing_type, levels=c("C>T","C>A","C>G"))) %>%
  ggplot( aes(x = adj_cytosine_position, y = editing, color = editing))+
  geom_jitter(size = .6, width = 0.1, alpha = 0.4)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))+
  ylim(0,80) + facet_grid(rows = vars(dinucleotide_context), cols = vars(editing_type))+
  theme_bw(base_size = 10)+
  labs(title = "F2X editing - NGG ONLY",
       xlab = "Cytosine position",
       ylab = "Percent editing",
       color = NULL)+ 
  labs(color = NULL)+ scale_color_viridis(option = "A", limits = c(0,80))
#FNLS_NG
HBES_FNLSNG_ALL_stack[HBES_FNLSNG_ALL_stack$PAM_class == c("NGG"),] %>% 
  mutate(editing_type = fct_relevel(editing_type, levels=c("C>T","C>A","C>G"))) %>%
  ggplot( aes(x = adj_cytosine_position, y = editing, color = editing))+
  geom_jitter(size = .6, width = 0.1, alpha = 0.4)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))+
  ylim(0,80) + facet_grid(rows = vars(dinucleotide_context), cols = vars(editing_type))+
  theme_bw(base_size = 10)+
  labs(title = "FNLS-NG editing - NGG ONLY",
       xlab = "Cytosine position",
       ylab = "Percent editing",
       color = NULL)+ 
  labs(color = NULL)+ scale_color_viridis(option = "A", limits = c(0,80))
```

Editing range across enzyme and PAM class

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=10,fig.height=6}
HBES_ALL_stack %>% 
  mutate(PAM_class = fct_relevel(PAM_class, levels=c("NGA","NGT","NGC","NGG","NAG","NNG","GAN"))) %>%
  mutate(enzyme = fct_relevel(enzyme, levels=c("FNLS","F2X","FNLSNG"))) %>%
  ggplot( aes(x = adj_cytosine_position, y = percent_tCTN, color = percent_tCTN))+
  geom_jitter(size = .25, width = 0.2, alpha = 1)+
  theme_bw(base_line_size = 0.25)+
  ylim(0,80) +
  labs(title = "C>T editing range (HBES-MDA231)",
       xlab = "Cytosine position",
       ylab = "Percent C>T editing",color = NULL)+
  scale_color_viridis(option = "A", limits = c(0,80))+
  facet_grid(rows = vars(enzyme), cols = vars(PAM_class))+
  ggsave("MDA231_HBES_tCTN_range.eps", dpi = 300, width = 25, height = 15, units = "cm")
```

Alternatively, average BE activity across different cytosine positions can be visualized by heatmap. In this case, the window is restricted to positions 1-20 of the protospacer, as this encapsulates almost all of the editing events. Can use this approach to view all BE events, or just C>T, C>G, C>A

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=10,fig.height=4}
# ALL BE
HBES_percent_BE_mean[HBES_percent_BE_mean$adj_cytosine_position >= 1 & HBES_percent_BE_mean$adj_cytosine_position <=20, ] %>%
  mutate(PAM_class = fct_relevel(PAM_class, levels=c("NGT","NGC","NGA","NGG","NAG","NNG","GAN"))) %>% 
  mutate(enzyme = fct_relevel(enzyme, levels=c("FNLS","F2X","FNLSNG"))) %>%
  mutate(dinucleotide_context = fct_relevel(dinucleotide_context, levels=c("AC","TC","CC","GC"))) %>%
  ggplot(aes(x = adj_cytosine_position, y = PAM_class, fill = percent_BE)) + 
  geom_tile() +
  theme_bw(base_size = 10)+
  scale_fill_viridis(option = "C", limits = c(0,100), na.value = "grey50")+
  facet_grid(rows = vars(enzyme), cols = vars(dinucleotide_context))
# C>T
HBES_percent_tCTN_mean[HBES_percent_tCTN_mean$adj_cytosine_position >= 1 & HBES_percent_tCTN_mean$adj_cytosine_position <=20, ] %>%
  mutate(PAM_class = fct_relevel(PAM_class, levels=c("NGT","NGC","NGA","NGG","NAG","NNG","GAN"))) %>% 
  mutate(enzyme = fct_relevel(enzyme, levels=c("FNLS","F2X","FNLSNG"))) %>%
  mutate(dinucleotide_context = fct_relevel(dinucleotide_context, levels=c("AC","TC","CC","GC"))) %>%
  ggplot(aes(x = adj_cytosine_position, y = PAM_class, fill = percent_tCTN)) + 
  geom_tile() +
  theme_bw(base_size = 10)+
  scale_fill_viridis(option = "C", limits = c(0,80), na.value = "grey50")+
  facet_grid(rows = vars(enzyme), cols = vars(dinucleotide_context))
# C>G
HBES_percent_tCGN_mean[HBES_percent_tCGN_mean$adj_cytosine_position >= 1 & HBES_percent_tCGN_mean$adj_cytosine_position <=20, ] %>%
  mutate(PAM_class = fct_relevel(PAM_class, levels=c("NGT","NGC","NGA","NGG","NAG","NNG","GAN"))) %>% 
  mutate(enzyme = fct_relevel(enzyme, levels=c("FNLS","F2X","FNLSNG"))) %>%
  mutate(dinucleotide_context = fct_relevel(dinucleotide_context, levels=c("AC","TC","CC","GC"))) %>%
  ggplot(aes(x = adj_cytosine_position, y = PAM_class, fill = percent_tCGN)) + 
  geom_tile() +
  theme_bw(base_size = 10)+
  scale_fill_viridis(option = "C", limits = c(0,15), na.value = "grey50")+
  facet_grid(rows = vars(enzyme), cols = vars(dinucleotide_context))
# C>A
HBES_percent_tCAN_mean[HBES_percent_tCAN_mean$adj_cytosine_position >= 1 & HBES_percent_tCAN_mean$adj_cytosine_position <=20, ] %>%
  mutate(PAM_class = fct_relevel(PAM_class, levels=c("NGT","NGC","NGA","NGG","NAG","NNG","GAN"))) %>% 
  mutate(enzyme = fct_relevel(enzyme, levels=c("FNLS","F2X","FNLSNG"))) %>%
  mutate(dinucleotide_context = fct_relevel(dinucleotide_context, levels=c("AC","TC","CC","GC"))) %>%
  ggplot(aes(x = adj_cytosine_position, y = PAM_class, fill = percent_tCAN)) + 
  geom_tile() +
  theme_bw(base_size = 10)+
  scale_fill_viridis(option = "C", limits = c(0,15), na.value = "grey50")+
  facet_grid(rows = vars(enzyme), cols = vars(dinucleotide_context))
```

Non-canonical editing by dinucleotide context

```{r echo=TRUE, message=FALSE, warning=FALSE, fig.width=8,fig.height=4.5}
# Boxplot with stats (editing type ratios)
# C>G / C>T editing ratio
HBES_ALL_stack_5percent_filter %>%
  mutate(dinucleotide_context = fct_relevel(dinucleotide_context, levels=c("AC","TC","CC","GC"))) %>%
  ggboxplot(x = "dinucleotide_context", y = "GvsT_ratio",
          color = "dinucleotide_context", 
          fill = "dinucleotide_context", 
          palette = "jco",
          alpha = 0.5,
          title = "C>G / C>T editing ratio (>5% C>T)",
          ylab = "C>G / C>T ratio",
          xlab = "dinucleotide context",
          ggtheme = theme_bw(base_size = 10))+
  facet_grid(cols = vars(enzyme))+
  stat_summary(fun.data = function(x) data.frame(y = c(-0.1),
                                                 label = paste(round(mean(x),2))), 
               geom="text", size = 3)+
    rotate_x_text()+
  stat_compare_means(comparisons = list(c("TC","AC"),c("TC","GC"),c("TC","CC"),
                                        c("AC","GC"),c("AC","CC"),c("GC","CC")))
# No stats
HBES_ALL_stack_5percent_filter %>%
  mutate(dinucleotide_context = fct_relevel(dinucleotide_context, levels=c("AC","TC","CC","GC"))) %>%
  ggboxplot(x = "dinucleotide_context", y = "GvsT_ratio",
          color = "dinucleotide_context", 
          fill = "dinucleotide_context", 
          palette = "jco",
          alpha = 0.5,
          title = "C>G / C>T editing ratio (>5% C>T)",
          ylab = "C>G / C>T ratio",
          xlab = "dinucleotide context",
          ggtheme = theme_bw(base_size = 10))+
  facet_grid(cols = vars(enzyme))+
  stat_summary(fun.data = function(x) data.frame(y = c(-0.1),
                                                 label = paste(round(mean(x),2))), 
               geom="text", size = 3)+
    rotate_x_text()

# C>A / C>T editing ratio
HBES_ALL_stack_5percent_filter %>%
  mutate(dinucleotide_context = fct_relevel(dinucleotide_context, levels=c("AC","TC","CC","GC"))) %>%
  ggboxplot(x = "dinucleotide_context",y = "AvsT_ratio",
          color = "dinucleotide_context", 
          fill = "dinucleotide_context", 
          palette = "jco",
          alpha = 0.5,
          title = "C>A / C>T editing ratio (>5% C>T)",
          ylab = "C>A / C>T ratio",
          xlab = "dinucleotide context",
          ggtheme = theme_bw(base_size = 10))+
  facet_grid(cols = vars(enzyme))+
  stat_summary(fun.data = function(x) data.frame(y = c(-0.1),
                                                 label = paste(round(mean(x),2))), 
               geom="text", size = 3)+
    rotate_x_text()+
  stat_compare_means(comparisons = list(c("TC","AC"),c("TC","GC"),c("TC","CC"),
                                        c("AC","GC"),c("AC","CC"),c("GC","CC")))
# No stats
HBES_ALL_stack_5percent_filter %>%
  mutate(dinucleotide_context = fct_relevel(dinucleotide_context, levels=c("AC","TC","CC","GC"))) %>%
  ggboxplot(x = "dinucleotide_context", y = "AvsT_ratio",
          color = "dinucleotide_context", 
          fill = "dinucleotide_context", 
          palette = "jco",
          alpha = 0.5,
          title = "C>A / C>T editing ratio (>5% C>T)",
          ylab = "C>A / C>T ratio",
          xlab = "dinucleotide context",
          ggtheme = theme_bw(base_size = 10))+
  facet_grid(cols = vars(enzyme))+
  stat_summary(fun.data = function(x) data.frame(y = c(-0.1),
                                                 label = paste(round(mean(x),2))), 
               geom="text", size = 3)+
    rotate_x_text()
```

